<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Data preparation Muraro (2016)</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span. { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">project-cell-type-annotation</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fas fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Preprocessing
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="dataprep_baron.html">Baron Dataset</a>
    </li>
    <li>
      <a href="dataprep_muraro.html">Muraro Dataset</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Methods
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="SingleR.html">SingleR</a>
    </li>
    <li>
      <a href="scClassify.html">scClassify</a>
    </li>
    <li>
      <a href="Seurat.html">Seurat</a>
    </li>
  </ul>
</li>
<li>
  <a href="evaluation.html">Evaluation</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Data preparation Muraro (2016)</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-01-06
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>project-cell-type-annotation/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsta426hs2020projectcelltypeannotationtree8f3f72adaf6ec6dd2a98fdaff4f01f4fa9e78f75targetblank8f3f72aa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sta426hs2020/project-cell-type-annotation/tree/8f3f72adaf6ec6dd2a98fdaff4f01f4fa9e78f75" target="_blank">8f3f72a</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsta426hs2020projectcelltypeannotationtree8f3f72adaf6ec6dd2a98fdaff4f01f4fa9e78f75targetblank8f3f72aa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sta426hs2020/project-cell-type-annotation/tree/8f3f72adaf6ec6dd2a98fdaff4f01f4fa9e78f75" target="_blank">8f3f72a</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    data/.DS_Store
    Ignored:    data/.Rapp.history

Untracked files:
    Untracked:  analysis/overview.Rmd

Unstaged changes:
    Modified:   analysis/_site.yml
    Deleted:    analysis/about.Rmd
    Modified:   data/Seurat.pred.baron.labels.rds
    Modified:   data/baron.labels.counts.rds
    Modified:   data/muraro.labels.counts.rds
    Modified:   data/scClassify.pred.muraro.labels.rds
    Modified:   data/sce_baron.RDS
    Modified:   data/sce_muraro.RDS
    Modified:   ref.bibtex

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/dataprep_muraro.Rmd</code>) and HTML (<code>docs/dataprep_muraro.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/8f3f72adaf6ec6dd2a98fdaff4f01f4fa9e78f75/analysis/dataprep_muraro.Rmd" target="_blank">8f3f72a</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-06
</td>
<td>
wflow_publish(“analysis/dataprep_muraro.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/2823d79b16dac9827a36a200729e68f5f42f1f0e/docs/dataprep_muraro.html" target="_blank">2823d79</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/164528fea9dfc6841d65414b06bd9df526c34b6d/analysis/dataprep_muraro.Rmd" target="_blank">164528f</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-06
</td>
<td>
wflow_publish(files = “analysis/dataprep_muraro.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/f51946750a14f00bc75db37fc8499aaa5ee5ed87/docs/dataprep_muraro.html" target="_blank">f519467</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/05f7cba57edeff67211129c15c0fbe3991417292/analysis/dataprep_muraro.Rmd" target="_blank">05f7cba</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-03
</td>
<td>
wflow_publish(all = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/ef0a2f20647ac7960f6e7ea0a2d7f01547b8d044/docs/dataprep_muraro.html" target="_blank">ef0a2f2</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/08330046fe7c4e814193c80811461857574a2802/analysis/dataprep_muraro.Rmd" target="_blank">0833004</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-03
</td>
<td>
update and clean up data prep files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/7b3b6e37202bdd4ee1122e5a48750b7254a76fa9/analysis/dataprep_muraro.Rmd" target="_blank">7b3b6e3</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-30
</td>
<td>
wflow_git_commit(all = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/543a43c4e52e57046078cc48380673460c376972/docs/dataprep_muraro.html" target="_blank">543a43c</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/42d96430fd6e5a660f2587f10e94454ff144f61b/analysis/dataprep_muraro.Rmd" target="_blank">42d9643</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-30
</td>
<td>
wflow_publish(all = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/241b8939ed4d926e7902b2f9c3635ddadd1b1725/docs/dataprep_muraro.html" target="_blank">241b893</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/6e8108091bae79e61d315336c4e5efb2e6b826a4/analysis/dataprep_muraro.Rmd" target="_blank">6e81080</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-28
</td>
<td>
include dim red
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/7566ddbaeaefb84f89d192940456343ce7ad1f63/docs/dataprep_muraro.html" target="_blank">7566ddb</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/caf6cd2760c6d80f6db6f8f30a21102b22c190bd/analysis/dataprep_muraro.Rmd" target="_blank">caf6cd2</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-28
</td>
<td>
remove NA labels
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/7bf452cd9aeef8b615e86cfd7529b8288b22ecfd/docs/dataprep_muraro.html" target="_blank">7bf452c</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/1e613a2e9c75da9699d2525e0ef7f97084ce6457/analysis/dataprep_muraro.Rmd" target="_blank">1e613a2</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-28
</td>
<td>
update dataprep
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/4afcd2606e33999c2d246f8b9d6dfc41d037ce95/docs/dataprep_muraro.html" target="_blank">4afcd26</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/c1caab9e502c323aa093b916d2e0a389af0a3c67/analysis/dataprep_muraro.Rmd" target="_blank">c1caab9</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-16
</td>
<td>
wflow_git_commit(“analysis/dataprep_muraro.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(scRNAseq))</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(AnnotationHub))</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(scater))</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(scran))</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(scuttle))</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(scDblFinder))</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(BiocParallel))</a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(knitr))</a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(kableExtra))</a></code></pre></div>
<div id="loading-the-data" class="section level1">
<h1>Loading the data</h1>
<p>We loaded the Muraro dataset directly from the <em>scRNAseq</em> package as a SingleCellExperiment object. The dataset contains CEL-seq scRNA-seq data from human pancreas cells from 4 different donors <span class="citation">(Muraro et al. 2016)</span>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">sce.muraro &lt;-<span class="st"> </span><span class="kw">suppressMessages</span>(<span class="kw">MuraroPancreasData</span>(<span class="dt">ensembl =</span> <span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw">dim</span>(sce.muraro)</a></code></pre></div>
<pre><code>[1] 16940  3072</code></pre>
<p>The option <strong>ensebml = TRUE</strong> ensures that the gene symbols are converted to Ensembl IDs. Any genes that could not be matched to an Ensembl ID were discarded (2110 genes). We were left with <span class="math inline">\(16940\)</span> genes from <span class="math inline">\(3072\)</span> different cells.</p>
</div>
<div id="quality-control" class="section level1">
<h1>Quality Control</h1>
<p>Quality control was performed on the loaded Baron dataset.</p>
<p>First we tried to identify cells with relatively high proportion of mitochondrial genes, which should be removed, as this indicates damaged nuclei, and thus low-quality cells.</p>
<div id="visual-inspection-before-qc" class="section level2">
<h2>Visual Inspection before QC</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">location &lt;-<span class="st"> </span><span class="kw">rowRanges</span>(sce.muraro)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">isMito &lt;-<span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^mt-&quot;</span>, location<span class="op">$</span>originalName, <span class="dt">ignore.case=</span><span class="ot">TRUE</span>) <span class="op">|</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="st">  </span><span class="kw">seqnames</span>(location)<span class="op">==</span><span class="st">&quot;MT&quot;</span> <span class="op">|</span><span class="st"> </span><span class="kw">seqnames</span>(location)<span class="op">==</span><span class="st">&quot;mt&quot;</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="kw">ifelse</span>(<span class="kw">any</span>(isMito), <span class="st">&quot;There has been found at least one mitochondrial gene&quot;</span>,</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">       <span class="st">&quot;There have not been found any mitochondrial genes&quot;</span>)</a></code></pre></div>
<pre><code>[1] &quot;There have not been found any mitochondrial genes&quot;</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">df &lt;-<span class="st"> </span><span class="kw">perCellQCMetrics</span>(sce.muraro)</a></code></pre></div>
<p>Suprisingly, we saw that there are no mitochondrial genes left in the dataset. We assume that the authors have removed all mitochondrial genes beforehand.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co">#get QC metrics:</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2">sce.muraro &lt;-<span class="st"> </span><span class="kw">addPerCellQC</span>(sce.muraro)</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">sce.muraro &lt;-<span class="st"> </span><span class="kw">addPerFeatureQC</span>(sce.muraro)</a>
<a class="sourceLine" id="cb7-4" data-line-number="4"></a>
<a class="sourceLine" id="cb7-5" data-line-number="5">qc &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">colData</span>(sce.muraro))</a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="kw">ggplot</span>(qc, <span class="kw">aes</span>(sum)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_histogram</span>() <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>donor) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">trans =</span> <span class="st">&#39;log10&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Libary sizes of every genes split by donor&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of genes&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Library Size&quot;</span>)</a></code></pre></div>
<p><img src="figure/dataprep_muraro.Rmd/plot1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot1-1">
Past versions of plot1-1.png
</button>
</p>
<div id="fig-plot1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/2823d79b16dac9827a36a200729e68f5f42f1f0e/docs/figure/dataprep_muraro.Rmd/plot1-1.png" target="_blank">2823d79</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The histograms of the library sizes show bimodal distributions, especially in donor D28. What we would like to have though, are unimodal, approximately normally distributed library sizes.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw">ggplot</span>(qc, <span class="kw">aes</span>(sum, detected)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_density2d</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of detected genes&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Library size&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">trans =</span> <span class="st">&#39;log10&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">trans =</span> <span class="st">&#39;log10&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Library size vs. number of detected genes per cell&quot;</span>)</a></code></pre></div>
<p><img src="figure/dataprep_muraro.Rmd/plot2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot2-1">
Past versions of plot2-1.png
</button>
</p>
<div id="fig-plot2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/2823d79b16dac9827a36a200729e68f5f42f1f0e/docs/figure/dataprep_muraro.Rmd/plot2-1.png" target="_blank">2823d79</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>As we can see from the plot, the range of library sizes as well as the range of the number of detected genes is too large.</p>
</div>
<div id="qc" class="section level2">
<h2>QC</h2>
<p>Thus, we have to perform quantitative outlier detection, in order to discard cells that have very low library size or a too low number of detected genes. Additionaly, we want to discard cells with a too high ERCC spike-in content, as this is symptomatic of loss of endogenous RNA <span class="citation">(Amezquita et al. 2020)</span>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">reasons &lt;-<span class="st"> </span><span class="kw">quickPerCellQC</span>(df, </a>
<a class="sourceLine" id="cb9-2" data-line-number="2">    <span class="dt">sub.fields=</span><span class="st">&quot;altexps_ERCC_percent&quot;</span>)</a>
<a class="sourceLine" id="cb9-3" data-line-number="3">discard_reasons &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">colSums</span>(<span class="kw">as.matrix</span>(reasons)))</a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="kw">names</span>(discard_reasons) &lt;-<span class="st"> &quot;counts&quot;</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5">knitr<span class="op">::</span><span class="kw">kable</span>(discard_reasons) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="st">  </span><span class="kw">kable_styling</span>()<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="st">  </span><span class="kw">row_spec</span>(<span class="dv">4</span>,<span class="dt">bold=</span>T)</a></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
counts
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
low_lib_size
</td>
<td style="text-align:right;">
611
</td>
</tr>
<tr>
<td style="text-align:left;">
low_n_features
</td>
<td style="text-align:right;">
669
</td>
</tr>
<tr>
<td style="text-align:left;">
high_altexps_ERCC_percent
</td>
<td style="text-align:right;">
696
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
discard
</td>
<td style="text-align:right;font-weight: bold;">
726
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="co">#spike-ins</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2">sce.muraro<span class="op">$</span>exclude_ERCC &lt;-<span class="st"> </span>reasons<span class="op">$</span>high_altexps_ERCC_percent</a>
<a class="sourceLine" id="cb10-3" data-line-number="3">sce.muraro<span class="op">$</span>discard &lt;-<span class="st"> </span>reasons<span class="op">$</span>discard</a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="kw">plotColData</span>(sce.muraro, <span class="dt">x=</span><span class="st">&quot;donor&quot;</span>, <span class="dt">y=</span><span class="st">&quot;altexps_ERCC_percent&quot;</span>, <span class="dt">colour_by =</span> <span class="kw">I</span>(reasons<span class="op">$</span>high_altexps_ERCC_percent)) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Donor&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Remove outliers based on ERCC spike-ins&quot;</span>)</a></code></pre></div>
<p><img src="figure/dataprep_muraro.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/2823d79b16dac9827a36a200729e68f5f42f1f0e/docs/figure/dataprep_muraro.Rmd/unnamed-chunk-1-1.png" target="_blank">2823d79</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/f51946750a14f00bc75db37fc8499aaa5ee5ed87/docs/figure/dataprep_muraro.Rmd/unnamed-chunk-1-1.png" target="_blank">f519467</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/ef0a2f20647ac7960f6e7ea0a2d7f01547b8d044/docs/figure/dataprep_muraro.Rmd/unnamed-chunk-1-1.png" target="_blank">ef0a2f2</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="co"># remove outliers</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2">sce.muraro &lt;-<span class="st"> </span>sce.muraro[,<span class="op">!</span><span class="kw">apply</span>(reasons, <span class="dt">MARGIN =</span> <span class="dv">1</span>, any)]</a></code></pre></div>
<p>From the table, we see that a total of <span class="math inline">\(726\)</span> cells were excluded. Further, we see that most cells were excluded for more than one reason. In total, <span class="math inline">\(611\)</span> cells were excluded due to low library size, <span class="math inline">\(669\)</span> are exluded due to low number of features and <span class="math inline">\(696\)</span> cells are excluded based on high spike-in counts.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="co"># get QC metrics:</span></a>
<a class="sourceLine" id="cb12-2" data-line-number="2">sce.muraro &lt;-<span class="st"> </span><span class="kw">addPerCellQC</span>(sce.muraro)</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">sce.muraro &lt;-<span class="st"> </span><span class="kw">addPerFeatureQC</span>(sce.muraro)</a></code></pre></div>
</div>
<div id="visual-inspection-after-qc" class="section level2">
<h2>Visual inspection after QC</h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="co"># plots</span></a>
<a class="sourceLine" id="cb13-2" data-line-number="2">qc &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">colData</span>(sce.muraro))</a>
<a class="sourceLine" id="cb13-3" data-line-number="3"></a>
<a class="sourceLine" id="cb13-4" data-line-number="4"><span class="kw">ggplot</span>(qc, <span class="kw">aes</span>(sum)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_histogram</span>() <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>donor) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Libary sizes of every genes split by donor&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">trans =</span> <span class="st">&quot;log10&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of genes&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;log10(Library Size)&quot;</span>)</a></code></pre></div>
<p><img src="figure/dataprep_muraro.Rmd/qc%20visual-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-qc-visual-1">
Past versions of “qc visual-1.png”
</button>
</p>
<div id="fig-no-spaces-qc-visual-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/2823d79b16dac9827a36a200729e68f5f42f1f0e/docs/figure/dataprep_muraro.Rmd/qc visual-1.png" target="_blank">2823d79</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Looking at the histogram, we see that the library sizes are now reasonably distributed. Additionally, we see that most cells that have been excluded, came from donor D28.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw">ggplot</span>(qc, <span class="kw">aes</span>(sum, detected)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_density2d</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of detected genes&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Library size)&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">trans =</span> <span class="st">&#39;log10&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">trans =</span> <span class="st">&#39;log10&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Library size vs. number of detected genes per cell&quot;</span>)</a></code></pre></div>
<p><img src="figure/dataprep_muraro.Rmd/plot3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot3-1">
Past versions of plot3-1.png
</button>
</p>
<div id="fig-plot3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/2823d79b16dac9827a36a200729e68f5f42f1f0e/docs/figure/dataprep_muraro.Rmd/plot3-1.png" target="_blank">2823d79</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>After removal of all outliers, we see that the range of library sizes, as well as the number of detected genes, is now much narrower than before.</p>
</div>
</div>
<div id="unifying-label-names" class="section level1">
<h1>Unifying Label Names</h1>
<p>In order to be able to compare the two datasets by <span class="citation">Muraro et al. (2016)</span> and <span class="citation">Baron et al. (2016)</span>, we need to ensure that they have the same cell type labels. Therefore we rename “duct” to “ductal and”pp&quot; to “gamma”, similarly performed by <span class="citation">Lin et al. (2019)</span>. Further, we remove all cells that are either labelled “unclear” or “mesenchymal”, because these label types are not present in the Baron dataset. We are left with 2022 cells, belonging to 8 different cell types.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">sce.muraro<span class="op">$</span>label[sce.muraro<span class="op">$</span>label<span class="op">==</span><span class="st">&quot;duct&quot;</span>] &lt;-<span class="st"> &quot;ductal&quot;</span> <span class="co">#renaming</span></a>
<a class="sourceLine" id="cb15-2" data-line-number="2">sce.muraro<span class="op">$</span>label[sce.muraro<span class="op">$</span>label<span class="op">==</span><span class="st">&quot;pp&quot;</span>] &lt;-<span class="st"> &quot;gamma&quot;</span> <span class="co">#renaming</span></a>
<a class="sourceLine" id="cb15-3" data-line-number="3">sce.muraro &lt;-<span class="st"> </span>sce.muraro[,<span class="op">!</span>(sce.muraro<span class="op">$</span>label <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="ot">NA</span>,<span class="st">&quot;unclear&quot;</span>, <span class="st">&quot;mesenchymal&quot;</span>))] <span class="co">#removing</span></a>
<a class="sourceLine" id="cb15-4" data-line-number="4">muraro.labels.counts &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">table</span>(sce.muraro<span class="op">$</span>label))</a>
<a class="sourceLine" id="cb15-5" data-line-number="5"><span class="kw">names</span>(muraro.labels.counts) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;cell type&quot;</span>, <span class="st">&quot;frequency&quot;</span>)</a>
<a class="sourceLine" id="cb15-6" data-line-number="6">total &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;total&quot;</span>, <span class="kw">sum</span>(muraro.labels.counts<span class="op">$</span>frequency))</a>
<a class="sourceLine" id="cb15-7" data-line-number="7"><span class="kw">names</span>(total) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;cell type&quot;</span>, <span class="st">&quot;frequency&quot;</span>)</a>
<a class="sourceLine" id="cb15-8" data-line-number="8">muraro.labels.counts &lt;-<span class="st"> </span><span class="kw">rbind</span>(muraro.labels.counts, total)</a>
<a class="sourceLine" id="cb15-9" data-line-number="9"><span class="kw">saveRDS</span>(muraro.labels.counts, <span class="dt">file=</span><span class="st">&quot;data/muraro.labels.counts.rds&quot;</span>)</a>
<a class="sourceLine" id="cb15-10" data-line-number="10">knitr<span class="op">::</span><span class="kw">kable</span>(muraro.labels.counts) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-11" data-line-number="11"><span class="st">  </span><span class="kw">kable_styling</span>()<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-12" data-line-number="12"><span class="st">  </span><span class="kw">row_spec</span>(<span class="dv">9</span>,<span class="dt">bold=</span>T)</a></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
cell type
</th>
<th style="text-align:right;">
frequency
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
acinar
</td>
<td style="text-align:right;">
218
</td>
</tr>
<tr>
<td style="text-align:left;">
alpha
</td>
<td style="text-align:right;">
803
</td>
</tr>
<tr>
<td style="text-align:left;">
beta
</td>
<td style="text-align:right;">
446
</td>
</tr>
<tr>
<td style="text-align:left;">
delta
</td>
<td style="text-align:right;">
191
</td>
</tr>
<tr>
<td style="text-align:left;">
ductal
</td>
<td style="text-align:right;">
242
</td>
</tr>
<tr>
<td style="text-align:left;">
endothelial
</td>
<td style="text-align:right;">
20
</td>
</tr>
<tr>
<td style="text-align:left;">
epsilon
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
gamma
</td>
<td style="text-align:right;">
99
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
total
</td>
<td style="text-align:right;font-weight: bold;">
2022
</td>
</tr>
</tbody>
</table>
</div>
<div id="tsne-plot" class="section level1">
<h1>tSNE plot</h1>
<p>Finally, we produce normalized counts and perform dimensionality reduction for visual inspection purposes.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">sce.muraro &lt;-<span class="st"> </span><span class="kw">logNormCounts</span>(sce.muraro)</a></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">sce.muraro &lt;-<span class="st"> </span><span class="kw">runPCA</span>(sce.muraro)</a>
<a class="sourceLine" id="cb17-2" data-line-number="2">sce.muraro &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(sce.muraro, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw">plotReducedDim</span>(sce.muraro, <span class="st">&quot;TSNE&quot;</span>, <span class="dt">colour_by=</span><span class="st">&quot;label&quot;</span>) <span class="co">#original labels</span></a></code></pre></div>
<p><img src="figure/dataprep_muraro.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/2823d79b16dac9827a36a200729e68f5f42f1f0e/docs/figure/dataprep_muraro.Rmd/unnamed-chunk-2-1.png" target="_blank">2823d79</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We can see that the “true” cell type labels correspond quite nicely to the clusters in the tSNE plot.</p>
<div id="save-object" class="section level2">
<h2>Save object</h2>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw">saveRDS</span>(sce.muraro,<span class="dt">file =</span> <span class="st">&quot;data/sce_muraro.rds&quot;</span>)</a></code></pre></div>
</div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs">
<div id="ref-amezquita2020orchestrating">
<p>Amezquita, Robert A, Aaron TL Lun, Etienne Becht, Vince J Carey, Lindsay N Carpp, Ludwig Geistlinger, Federico Marini, et al. 2020. “Orchestrating Single-Cell Analysis with Bioconductor.” <em>Nature Methods</em> 17 (2). Nature Publishing Group: 137–45.</p>
</div>
<div id="ref-baron2016single">
<p>Baron, Maayan, Adrian Veres, Samuel L Wolock, Aubrey L Faust, Renaud Gaujoux, Amedeo Vetere, Jennifer Hyoje Ryu, et al. 2016. “A Single-Cell Transcriptomic Map of the Human and Mouse Pancreas Reveals Inter-and Intra-Cell Population Structure.” <em>Cell Systems</em> 3 (4). Elsevier: 346–60.</p>
</div>
<div id="ref-lin2019scclassify">
<p>Lin, Yingxin, Yao Cao, Hani J Kim, Agus Salim, Terence P Speed, Dave Lin, Pengyi Yang, and Jean Yee Hwa Yang. 2019. “ScClassify: Hierarchical Classification of Cells.” <em>bioRxiv</em>. Cold Spring Harbor Laboratory, 776948.</p>
</div>
<div id="ref-muraro2016single">
<p>Muraro, Mauro J, Gitanjali Dharmadhikari, Dominic Grün, Nathalie Groen, Tim Dielen, Erik Jansen, Leon van Gurp, et al. 2016. “A Single-Cell Transcriptome Atlas of the Human Pancreas.” <em>Cell Systems</em> 3 (4). Elsevier: 385–94.</p>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ensembldb_2.14.0            AnnotationFilter_1.14.0    
 [3] GenomicFeatures_1.42.1      AnnotationDbi_1.52.0       
 [5] kableExtra_1.3.1            knitr_1.30                 
 [7] BiocParallel_1.24.1         scDblFinder_1.4.0          
 [9] scuttle_1.0.4               scran_1.18.3               
[11] scater_1.18.3               ggplot2_3.3.2              
[13] AnnotationHub_2.22.0        BiocFileCache_1.14.0       
[15] dbplyr_2.0.0                scRNAseq_2.4.0             
[17] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0
[19] Biobase_2.50.0              GenomicRanges_1.42.0       
[21] GenomeInfoDb_1.26.2         IRanges_2.24.1             
[23] S4Vectors_0.28.1            BiocGenerics_0.36.0        
[25] MatrixGenerics_1.2.0        matrixStats_0.57.0         
[27] workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] igraph_1.2.6                  lazyeval_0.2.2               
  [3] digest_0.6.27                 htmltools_0.5.0              
  [5] viridis_0.5.1                 magrittr_2.0.1               
  [7] memoise_1.1.0                 limma_3.46.0                 
  [9] Biostrings_2.58.0             askpass_1.1                  
 [11] prettyunits_1.1.1             colorspace_2.0-0             
 [13] blob_1.2.1                    rvest_0.3.6                  
 [15] rappdirs_0.3.1                xfun_0.19                    
 [17] dplyr_1.0.2                   crayon_1.3.4                 
 [19] RCurl_1.98-1.2                glue_1.4.2                   
 [21] gtable_0.3.0                  zlibbioc_1.36.0              
 [23] XVector_0.30.0                webshot_0.5.2                
 [25] DelayedArray_0.16.0           BiocSingular_1.6.0           
 [27] scales_1.1.1                  DBI_1.1.0                    
 [29] edgeR_3.32.0                  Rcpp_1.0.5                   
 [31] isoband_0.2.3                 viridisLite_0.3.0            
 [33] xtable_1.8-4                  progress_1.2.2               
 [35] dqrng_0.2.1                   bit_4.0.4                    
 [37] rsvd_1.0.3                    httr_1.4.2                   
 [39] ellipsis_0.3.1                pkgconfig_2.0.3              
 [41] XML_3.99-0.5                  farver_2.0.3                 
 [43] locfit_1.5-9.4                tidyselect_1.1.0             
 [45] labeling_0.4.2                rlang_0.4.9                  
 [47] later_1.1.0.1                 munsell_0.5.0                
 [49] BiocVersion_3.12.0            tools_4.0.3                  
 [51] xgboost_1.2.0.1               generics_0.1.0               
 [53] RSQLite_2.2.1                 ExperimentHub_1.16.0         
 [55] evaluate_0.14                 stringr_1.4.0                
 [57] fastmap_1.0.1                 yaml_2.2.1                   
 [59] bit64_4.0.5                   fs_1.5.0                     
 [61] purrr_0.3.4                   sparseMatrixStats_1.2.0      
 [63] whisker_0.4                   mime_0.9                     
 [65] xml2_1.3.2                    biomaRt_2.46.0               
 [67] compiler_4.0.3                rstudioapi_0.13              
 [69] beeswarm_0.2.3                curl_4.3                     
 [71] interactiveDisplayBase_1.28.0 tibble_3.0.4                 
 [73] statmod_1.4.35                stringi_1.5.3                
 [75] highr_0.8                     lattice_0.20-41              
 [77] bluster_1.0.0                 ProtGenerics_1.22.0          
 [79] Matrix_1.3-0                  vctrs_0.3.6                  
 [81] pillar_1.4.7                  lifecycle_0.2.0              
 [83] BiocManager_1.30.10           BiocNeighbors_1.8.2          
 [85] cowplot_1.1.0                 data.table_1.13.4            
 [87] bitops_1.0-6                  irlba_2.3.3                  
 [89] httpuv_1.5.4                  rtracklayer_1.50.0           
 [91] R6_2.5.0                      promises_1.1.1               
 [93] gridExtra_2.3                 vipor_0.4.5                  
 [95] MASS_7.3-53                   assertthat_0.2.1             
 [97] openssl_1.4.3                 rprojroot_2.0.2              
 [99] withr_2.3.0                   GenomicAlignments_1.26.0     
[101] Rsamtools_2.6.0               GenomeInfoDbData_1.2.4       
[103] hms_0.5.3                     grid_4.0.3                   
[105] beachmat_2.6.4                rmarkdown_2.6                
[107] DelayedMatrixStats_1.12.1     Rtsne_0.15                   
[109] git2r_0.27.1                  shiny_1.5.0                  
[111] ggbeeswarm_0.6.0             </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
