<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>felix_first_tries</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">project-cell-type-annotation</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="felix_first_tries.html">Felix_tries</a>
</li>
<li>
  <a href="SingleR.html">SingleR</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">felix_first_tries</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-12-16
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>project-cell-type-annotation/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsta426hs2020projectcelltypeannotationtree407d2a95f7b855029ef306830437d987919dfe1btargetblank407d2a9a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sta426hs2020/project-cell-type-annotation/tree/407d2a95f7b855029ef306830437d987919dfe1b" target="_blank">407d2a9</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsta426hs2020projectcelltypeannotationtree407d2a95f7b855029ef306830437d987919dfe1btargetblank407d2a9a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sta426hs2020/project-cell-type-annotation/tree/407d2a95f7b855029ef306830437d987919dfe1b" target="_blank">407d2a9</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/felix_first_tries.Rmd</code>) and HTML (<code>docs/felix_first_tries.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/407d2a95f7b855029ef306830437d987919dfe1b/analysis/felix_first_tries.Rmd" target="_blank">407d2a9</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-16
</td>
<td>
resolve some conflict
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/4c83a03320797aca6fd12953349fc34f45e1a610/docs/felix_first_tries.html" target="_blank">4c83a03</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/43166ea96b8a0d009cdb6bccdc4aa27d935a9942/analysis/felix_first_tries.Rmd" target="_blank">43166ea</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-16
</td>
<td>
change yaml header
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/06a969e2b752c5d0f1dc977e462174ddb6fd9d45/docs/felix_first_tries.html" target="_blank">06a969e</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/97c972e80547ceba0e2ec36bd331f12a448e1153/analysis/felix_first_tries.Rmd" target="_blank">97c972e</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-10
</td>
<td>
updated readme
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/97c972e80547ceba0e2ec36bd331f12a448e1153/docs/felix_first_tries.html" target="_blank">97c972e</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-10
</td>
<td>
updated readme
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/da9438090bbeb39d118e3fc46592df138d237467/docs/felix_first_tries.html" target="_blank">da94380</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/7fa51711e8fbfde4c5b2592b10aad2a511efafaa/analysis/felix_first_tries.Rmd" target="_blank">7fa5171</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-10
</td>
<td>
first tries with singleR and scCLassify
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="loading-ref-dataset" class="section level2">
<h2>Loading ref dataset</h2>
<pre class="r"><code>#library(devtools)
#devtools::install_github(&quot;SydneyBioX/scClassify&quot;)
library(scClassify)


suppressPackageStartupMessages(library(&quot;celldex&quot;))
suppressPackageStartupMessages(library(SingleCellExperiment))
suppressPackageStartupMessages(library(SingleR))</code></pre>
<pre class="r"><code>ref &lt;- celldex::HumanPrimaryCellAtlasData(ensembl=TRUE)</code></pre>
<pre><code>snapshotDate(): 2020-10-02</code></pre>
<pre><code>see ?celldex and browseVignettes(&#39;celldex&#39;) for documentation</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>see ?celldex and browseVignettes(&#39;celldex&#39;) for documentation</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>snapshotDate(): 2020-10-27</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>require(&quot;ensembldb&quot;)</code></pre>
<pre><code>Warning: Unable to map 1470 of 19363 requested IDs.</code></pre>
<pre class="r"><code>ref</code></pre>
<pre><code>class: SummarizedExperiment 
dim: 17893 713 
metadata(0):
assays(1): logcounts
rownames(17893): ENSG00000121410 ENSG00000268895 ... ENSG00000159840
  ENSG00000074755
rowData names(0):
colnames(713): GSM112490 GSM112491 ... GSM92233 GSM92234
colData names(3): label.main label.fine label.ont</code></pre>
<pre class="r"><code>table(colData(ref)$label.main)</code></pre>
<pre><code>
           Astrocyte               B_cell                   BM 
                   2                   26                    7 
          BM &amp; Prog.         Chondrocytes                  CMP 
                   1                    8                    2 
                  DC Embryonic_stem_cells    Endothelial_cells 
                  88                   17                   64 
    Epithelial_cells         Erythroblast          Fibroblasts 
                  16                    8                   10 
         Gametocytes                  GMP          Hepatocytes 
                   5                    2                    3 
          HSC_-G-CSF            HSC_CD34+            iPS_cells 
                  10                    6                   42 
       Keratinocytes           Macrophage                  MEP 
                  25                   90                    2 
            Monocyte                  MSC            Myelocyte 
                  60                    9                    2 
Neuroepithelial_cell              Neurons          Neutrophils 
                   1                   16                   21 
             NK_cell          Osteoblasts            Platelets 
                   5                   15                    5 
    Pre-B_cell_CD34-     Pro-B_cell_CD34+        Pro-Myelocyte 
                   2                    2                    2 
 Smooth_muscle_cells              T_cells    Tissue_stem_cells 
                  16                   68                   55 </code></pre>
<pre class="r"><code>suppressPackageStartupMessages(library(TENxPBMCData))

tenx_pbmc4k &lt;- TENxPBMCData(dataset = &quot;pbmc4k&quot;)</code></pre>
<pre><code>snapshotDate(): 2020-10-02</code></pre>
<pre><code>see ?TENxPBMCData and browseVignettes(&#39;TENxPBMCData&#39;) for documentation</code></pre>
<pre><code>loading from cache</code></pre>
<pre class="r"><code>tenx_pbmc4k</code></pre>
<pre><code>class: SingleCellExperiment 
dim: 33694 4340 
metadata(0):
assays(1): counts
rownames(33694): ENSG00000243485 ENSG00000237613 ... ENSG00000277475
  ENSG00000268674
rowData names(3): ENSEMBL_ID Symbol_TENx Symbol
colnames: NULL
colData names(11): Sample Barcode ... Individual Date_published
reducedDimNames(0):
altExpNames(0):</code></pre>
</div>
<div id="prediction-using-singler" class="section level2">
<h2>prediction using SingleR</h2>
<pre class="r"><code>predictions &lt;- SingleR(test=tenx_pbmc4k, assay.type.test=1, 
    ref=ref, labels=ref$label.main)
table(predictions$labels)</code></pre>
<pre><code>
          B_cell              CMP               DC              GMP 
             606                8                1                2 
        Monocyte          NK_cell        Platelets Pre-B_cell_CD34- 
            1164              217                3               46 
         T_cells 
            2293 </code></pre>
</div>
<div id="prediction-using-scclassify" class="section level2">
<h2>prediction using scClassify</h2>
<pre class="r"><code>library(scran)
library(scater)</code></pre>
<pre><code>Lade nötiges Paket: ggplot2</code></pre>
<pre class="r"><code>library(scuttle)


tenx_pbmc4k &lt;- logNormCounts(tenx_pbmc4k)

## only select test cells with labels that appear at least 5 times in the reference dataset
col_index &lt;- colData(ref)$label.main %in% names(which(table(as.factor(colData(ref)$label.main)) &gt; 5))
ref_subset &lt;- ref[,col_index]
colnames(tenx_pbmc4k) &lt;- seq(1,dim(tenx_pbmc4k)[2])



scClassify_test &lt;- scClassify(exprsMat_train = assay(ref_subset, &quot;logcounts&quot;),
                                      cellTypes_train = colData(ref_subset)$label.main,
                                      exprsMat_test = assay(tenx_pbmc4k, &quot;logcounts&quot;),
                                      tree = &quot;HOPACH&quot;,
                                      algorithm = &quot;WKNN&quot;,
                                      selectFeatures = c(&quot;limma&quot;),
                                      similarity = &quot;pearson&quot;,
                                      weighted_ensemble = FALSE,
                                      returnList = FALSE,
                                      verbose = TRUE)</code></pre>
<pre><code>Ensemble learning is disabled... 
after filtering not expressed genes 
[1] 17893   677
[1] &quot;Feature Selection...&quot;
[1] &quot;Number of genes selected to construct HOPACH tree 1027&quot;
[1] &quot;Constructing tree ...&quot;
[1] &quot;Training....&quot;
[1] &quot;=== selecting features by: limma ====&quot;
Predicting using followings parameter combinations: 
     similarity algorithm features
[1,] &quot;pearson&quot;  &quot;WKNN&quot;    &quot;limma&quot; 
==================  Start classifying on test dataset  ==== 
Ensemble learning is disabled... 
Using parameters: 
similarity  algorithm   features 
 &quot;pearson&quot;     &quot;WKNN&quot;    &quot;limma&quot; </code></pre>
<pre><code>There are only 0 selected genes in reference data expressed in query data</code></pre>
<pre><code>weights for each base method: 
NULL</code></pre>
<pre class="r"><code>summary(rownames(tenx_pbmc4k) %in% rownames(ref_subset))</code></pre>
<pre><code>   Mode   FALSE    TRUE 
logical   16283   17411 </code></pre>
<div id="single-cell-pancreas-dataset-wang-et-al." class="section level3">
<h3>single cell pancreas dataset wang et al.</h3>
<pre class="r"><code>data(&quot;scClassify_example&quot;)
wang_cellTypes &lt;- scClassify_example$wang_cellTypes
exprsMat_wang_subset &lt;- scClassify_example$exprsMat_wang_subset
exprsMat_wang_subset &lt;- as(exprsMat_wang_subset, &quot;dgCMatrix&quot;)



data(&quot;trainClassExample_xin&quot;)
trainClassExample_xin</code></pre>
<pre><code>Class: scClassifyTrainModel 
Model name: training 
Feature selection methods: limma 
Number of cells in the training data: 674 
Number of cell types in the training data: 4 </code></pre>
<pre class="r"><code>scClassify::features(trainClassExample_xin)</code></pre>
<pre><code>[1] &quot;limma&quot;</code></pre>
<pre class="r"><code>plotCellTypeTree(cellTypeTree(trainClassExample_xin))</code></pre>
<pre><code>Warning: Removed 3 rows containing missing values (geom_text).</code></pre>
<p><img src="figure/felix_first_tries.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/97c972e80547ceba0e2ec36bd331f12a448e1153/docs/figure/felix_first_tries.Rmd/unnamed-chunk-5-1.png" target="_blank">97c972e</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/da9438090bbeb39d118e3fc46592df138d237467/docs/figure/felix_first_tries.Rmd/unnamed-chunk-5-1.png" target="_blank">da94380</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>pred_res &lt;- predict_scClassify(exprsMat_test = exprsMat_wang_subset,
                               trainRes = trainClassExample_xin,
                               cellTypes_test = wang_cellTypes,
                               algorithm = &quot;WKNN&quot;,
                               features = c(&quot;limma&quot;),
                               similarity = c(&quot;pearson&quot;, &quot;spearman&quot;),
                               prob_threshold = 0.7,
                               verbose = TRUE)</code></pre>
<pre><code>Performing unweighted ensemble learning... 
Using parameters: 
similarity  algorithm   features 
 &quot;pearson&quot;     &quot;WKNN&quot;    &quot;limma&quot; 
[1] &quot;Using dynamic correlation cutoff...&quot;
[1] &quot;Using dynamic correlation cutoff...&quot;
classify_res
               correct   correctly unassigned           intermediate 
           0.704590818            0.239520958            0.000000000 
incorrectly unassigned         error assigned          misclassified 
           0.000000000            0.051896208            0.003992016 
Using parameters: 
similarity  algorithm   features 
&quot;spearman&quot;     &quot;WKNN&quot;    &quot;limma&quot; 
[1] &quot;Using dynamic correlation cutoff...&quot;
[1] &quot;Using dynamic correlation cutoff...&quot;
classify_res
               correct   correctly unassigned           intermediate 
           0.702594810            0.013972056            0.000000000 
incorrectly unassigned         error assigned          misclassified 
           0.001996008            0.277445110            0.003992016 
weights for each base method: 
[1] NA NA</code></pre>
<pre class="r"><code>table(pred_res$pearson_WKNN_limma$predRes, wang_cellTypes)</code></pre>
<pre><code>                  wang_cellTypes
                   acinar alpha beta delta ductal gamma stellate
  alpha                 0   206    0     0      0     2        0
  beta                  0     0  118     0      1     0        0
  beta_delta_gamma      0     0    0     0     25     0        0
  delta                 0     0    0    10      0     0        0
  gamma                 0     0    0     0      0    19        0
  unassigned            5     0    0     0     70     0       45</code></pre>
<pre class="r"><code>table(pred_res$spearman_WKNN_limma$predRes, wang_cellTypes)</code></pre>
<pre><code>                  wang_cellTypes
                   acinar alpha beta delta ductal gamma stellate
  alpha                 0   206    0     0      0     2        2
  beta                  2     0  118     0     29     0        6
  beta_delta_gamma      1     0    0     0     66     0       31
  delta                 0     0    0    10      0     0        2
  gamma                 0     0    0     0      0    18        0
  unassigned            2     0    0     0      1     1        4</code></pre>
<pre class="r"><code>#install.packages(&#39;Seurat&#39;)
#library(Seurat)</code></pre>
<!-- ###  -->
<!-- good webpage: https://hemberg-lab.github.io/scRNA.seq.datasets/human/pancreas/ -->
<pre class="r"><code>BiocManager::version()</code></pre>
<pre><code>[1] &#39;3.12&#39;</code></pre>
<pre class="r"><code>BiocManager::valid()</code></pre>
<pre><code>Warning: 16 packages out-of-date; 1 packages too new</code></pre>
<pre><code>
* sessionInfo()

R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] de_CH.UTF-8/de_CH.UTF-8/de_CH.UTF-8/C/de_CH.UTF-8/de_CH.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] scuttle_1.0.3               scater_1.18.3              
 [3] ggplot2_3.3.2               scran_1.18.1               
 [5] TENxPBMCData_1.8.0          HDF5Array_1.18.0           
 [7] rhdf5_2.34.0                DelayedArray_0.16.0        
 [9] Matrix_1.2-18               ensembldb_2.14.0           
[11] AnnotationFilter_1.14.0     GenomicFeatures_1.42.1     
[13] AnnotationDbi_1.52.0        SingleR_1.4.0              
[15] SingleCellExperiment_1.12.0 celldex_1.0.0              
[17] SummarizedExperiment_1.20.0 Biobase_2.50.0             
[19] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
[21] IRanges_2.24.0              S4Vectors_0.28.0           
[23] BiocGenerics_0.36.0         MatrixGenerics_1.2.0       
[25] matrixStats_0.57.0          scClassify_1.3.1           
[27] workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] AnnotationHub_2.22.0          BiocFileCache_1.14.0         
  [3] igraph_1.2.6                  lazyeval_0.2.2               
  [5] proxyC_0.1.5                  splines_4.0.3                
  [7] BiocParallel_1.24.1           digest_0.6.27                
  [9] htmltools_0.5.0               viridis_0.5.1                
 [11] magrittr_2.0.1                memoise_1.1.0                
 [13] cluster_2.1.0                 mixtools_1.2.0               
 [15] limma_3.46.0                  Biostrings_2.58.0            
 [17] graphlayouts_0.7.1            RcppParallel_5.0.2           
 [19] askpass_1.1                   prettyunits_1.1.1            
 [21] colorspace_2.0-0              blob_1.2.1                   
 [23] rappdirs_0.3.1                ggrepel_0.8.2                
 [25] xfun_0.19                     dplyr_1.0.2                  
 [27] crayon_1.3.4                  RCurl_1.98-1.2               
 [29] survival_3.2-7                glue_1.4.2                   
 [31] polyclip_1.10-0               gtable_0.3.0                 
 [33] zlibbioc_1.36.0               XVector_0.30.0               
 [35] BiocSingular_1.6.0            kernlab_0.9-29               
 [37] Rhdf5lib_1.12.0               scales_1.1.1                 
 [39] edgeR_3.32.0                  DBI_1.1.0                    
 [41] Rcpp_1.0.5                    viridisLite_0.3.0            
 [43] xtable_1.8-4                  progress_1.2.2               
 [45] dqrng_0.2.1                   bit_4.0.4                    
 [47] rsvd_1.0.3                    proxy_0.4-24                 
 [49] httr_1.4.2                    ellipsis_0.3.1               
 [51] pkgconfig_2.0.3               XML_3.99-0.5                 
 [53] farver_2.0.3                  dbplyr_2.0.0                 
 [55] locfit_1.5-9.4                labeling_0.4.2               
 [57] tidyselect_1.1.0              rlang_0.4.9                  
 [59] later_1.1.0.1                 munsell_0.5.0                
 [61] BiocVersion_3.12.0            tools_4.0.3                  
 [63] generics_0.1.0                RSQLite_2.2.1                
 [65] ExperimentHub_1.16.0          evaluate_0.14                
 [67] stringr_1.4.0                 fastmap_1.0.1                
 [69] yaml_2.2.1                    knitr_1.30                   
 [71] bit64_4.0.5                   fs_1.5.0                     
 [73] tidygraph_1.2.0               purrr_0.3.4                  
 [75] ggraph_2.0.4                  nlme_3.1-150                 
 [77] sparseMatrixStats_1.2.0       whisker_0.4                  
 [79] mime_0.9                      xml2_1.3.2                   
 [81] biomaRt_2.46.0                compiler_4.0.3               
 [83] rstudioapi_0.13               beeswarm_0.2.3               
 [85] curl_4.3                      interactiveDisplayBase_1.28.0
 [87] tibble_3.0.4                  statmod_1.4.35               
 [89] tweenr_1.0.1                  stringi_1.5.3                
 [91] bluster_1.0.0                 lattice_0.20-41              
 [93] hopach_2.50.0                 ProtGenerics_1.22.0          
 [95] vctrs_0.3.5                   rhdf5filters_1.2.0           
 [97] pillar_1.4.7                  lifecycle_0.2.0              
 [99] BiocManager_1.30.10           BiocNeighbors_1.8.2          
[101] bitops_1.0-6                  irlba_2.3.3                  
[103] httpuv_1.5.4                  rtracklayer_1.50.0           
[105] R6_2.5.0                      promises_1.1.1               
[107] gridExtra_2.3                 vipor_0.4.5                  
[109] MASS_7.3-53                   assertthat_0.2.1             
[111] openssl_1.4.3                 rprojroot_2.0.2              
[113] minpack.lm_1.2-1              withr_2.3.0                  
[115] GenomicAlignments_1.26.0      Rsamtools_2.6.0              
[117] GenomeInfoDbData_1.2.4        diptest_0.75-7               
[119] mgcv_1.8-33                   hms_0.5.3                    
[121] grid_4.0.3                    beachmat_2.6.2               
[123] tidyr_1.1.2                   rmarkdown_2.5                
[125] DelayedMatrixStats_1.12.1     segmented_1.3-0              
[127] git2r_0.27.1                  ggforce_0.3.2                
[129] shiny_1.5.0                   ggbeeswarm_0.6.0             

Bioconductor version &#39;3.12&#39;

  * 16 packages out-of-date
  * 1 packages too new

create a valid installation with

  BiocManager::install(c(
    &quot;beachmat&quot;, &quot;GetoptLong&quot;, &quot;htmlwidgets&quot;, &quot;IRanges&quot;, &quot;parallelly&quot;,
    &quot;pkgbuild&quot;, &quot;RcppAnnoy&quot;, &quot;rmarkdown&quot;, &quot;S4Vectors&quot;, &quot;scClassify&quot;, &quot;scran&quot;,
    &quot;sctransform&quot;, &quot;segmented&quot;, &quot;Seurat&quot;, &quot;spatstat.data&quot;, &quot;tinytex&quot;, &quot;uwot&quot;
  ), update = TRUE, ask = FALSE)

more details: BiocManager::valid()$too_new, BiocManager::valid()$out_of_date</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] de_CH.UTF-8/de_CH.UTF-8/de_CH.UTF-8/C/de_CH.UTF-8/de_CH.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] scuttle_1.0.3               scater_1.18.3              
 [3] ggplot2_3.3.2               scran_1.18.1               
 [5] TENxPBMCData_1.8.0          HDF5Array_1.18.0           
 [7] rhdf5_2.34.0                DelayedArray_0.16.0        
 [9] Matrix_1.2-18               ensembldb_2.14.0           
[11] AnnotationFilter_1.14.0     GenomicFeatures_1.42.1     
[13] AnnotationDbi_1.52.0        SingleR_1.4.0              
[15] SingleCellExperiment_1.12.0 celldex_1.0.0              
[17] SummarizedExperiment_1.20.0 Biobase_2.50.0             
[19] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
[21] IRanges_2.24.0              S4Vectors_0.28.0           
[23] BiocGenerics_0.36.0         MatrixGenerics_1.2.0       
[25] matrixStats_0.57.0          scClassify_1.3.1           
[27] workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] AnnotationHub_2.22.0          BiocFileCache_1.14.0         
  [3] igraph_1.2.6                  lazyeval_0.2.2               
  [5] proxyC_0.1.5                  splines_4.0.3                
  [7] BiocParallel_1.24.1           digest_0.6.27                
  [9] htmltools_0.5.0               viridis_0.5.1                
 [11] magrittr_2.0.1                memoise_1.1.0                
 [13] cluster_2.1.0                 mixtools_1.2.0               
 [15] limma_3.46.0                  Biostrings_2.58.0            
 [17] graphlayouts_0.7.1            RcppParallel_5.0.2           
 [19] askpass_1.1                   prettyunits_1.1.1            
 [21] colorspace_2.0-0              blob_1.2.1                   
 [23] rappdirs_0.3.1                ggrepel_0.8.2                
 [25] xfun_0.19                     dplyr_1.0.2                  
 [27] crayon_1.3.4                  RCurl_1.98-1.2               
 [29] survival_3.2-7                glue_1.4.2                   
 [31] polyclip_1.10-0               gtable_0.3.0                 
 [33] zlibbioc_1.36.0               XVector_0.30.0               
 [35] BiocSingular_1.6.0            kernlab_0.9-29               
 [37] Rhdf5lib_1.12.0               scales_1.1.1                 
 [39] edgeR_3.32.0                  DBI_1.1.0                    
 [41] Rcpp_1.0.5                    viridisLite_0.3.0            
 [43] xtable_1.8-4                  progress_1.2.2               
 [45] dqrng_0.2.1                   bit_4.0.4                    
 [47] rsvd_1.0.3                    proxy_0.4-24                 
 [49] httr_1.4.2                    ellipsis_0.3.1               
 [51] pkgconfig_2.0.3               XML_3.99-0.5                 
 [53] farver_2.0.3                  dbplyr_2.0.0                 
 [55] locfit_1.5-9.4                labeling_0.4.2               
 [57] tidyselect_1.1.0              rlang_0.4.9                  
 [59] later_1.1.0.1                 munsell_0.5.0                
 [61] BiocVersion_3.12.0            tools_4.0.3                  
 [63] generics_0.1.0                RSQLite_2.2.1                
 [65] ExperimentHub_1.16.0          evaluate_0.14                
 [67] stringr_1.4.0                 fastmap_1.0.1                
 [69] yaml_2.2.1                    knitr_1.30                   
 [71] bit64_4.0.5                   fs_1.5.0                     
 [73] tidygraph_1.2.0               purrr_0.3.4                  
 [75] ggraph_2.0.4                  nlme_3.1-150                 
 [77] sparseMatrixStats_1.2.0       whisker_0.4                  
 [79] mime_0.9                      xml2_1.3.2                   
 [81] biomaRt_2.46.0                compiler_4.0.3               
 [83] rstudioapi_0.13               beeswarm_0.2.3               
 [85] curl_4.3                      interactiveDisplayBase_1.28.0
 [87] tibble_3.0.4                  statmod_1.4.35               
 [89] tweenr_1.0.1                  stringi_1.5.3                
 [91] bluster_1.0.0                 lattice_0.20-41              
 [93] hopach_2.50.0                 ProtGenerics_1.22.0          
 [95] vctrs_0.3.5                   rhdf5filters_1.2.0           
 [97] pillar_1.4.7                  lifecycle_0.2.0              
 [99] BiocManager_1.30.10           BiocNeighbors_1.8.2          
[101] bitops_1.0-6                  irlba_2.3.3                  
[103] httpuv_1.5.4                  rtracklayer_1.50.0           
[105] R6_2.5.0                      promises_1.1.1               
[107] gridExtra_2.3                 vipor_0.4.5                  
[109] MASS_7.3-53                   assertthat_0.2.1             
[111] openssl_1.4.3                 rprojroot_2.0.2              
[113] minpack.lm_1.2-1              withr_2.3.0                  
[115] GenomicAlignments_1.26.0      Rsamtools_2.6.0              
[117] GenomeInfoDbData_1.2.4        diptest_0.75-7               
[119] mgcv_1.8-33                   hms_0.5.3                    
[121] grid_4.0.3                    beachmat_2.6.2               
[123] tidyr_1.1.2                   rmarkdown_2.5                
[125] DelayedMatrixStats_1.12.1     segmented_1.3-0              
[127] git2r_0.27.1                  ggforce_0.3.2                
[129] shiny_1.5.0                   ggbeeswarm_0.6.0             </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] de_CH.UTF-8/de_CH.UTF-8/de_CH.UTF-8/C/de_CH.UTF-8/de_CH.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] scuttle_1.0.3               scater_1.18.3              
 [3] ggplot2_3.3.2               scran_1.18.1               
 [5] TENxPBMCData_1.8.0          HDF5Array_1.18.0           
 [7] rhdf5_2.34.0                DelayedArray_0.16.0        
 [9] Matrix_1.2-18               ensembldb_2.14.0           
[11] AnnotationFilter_1.14.0     GenomicFeatures_1.42.1     
[13] AnnotationDbi_1.52.0        SingleR_1.4.0              
[15] SingleCellExperiment_1.12.0 celldex_1.0.0              
[17] SummarizedExperiment_1.20.0 Biobase_2.50.0             
[19] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
[21] IRanges_2.24.0              S4Vectors_0.28.0           
[23] BiocGenerics_0.36.0         MatrixGenerics_1.2.0       
[25] matrixStats_0.57.0          scClassify_1.3.1           
[27] workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] AnnotationHub_2.22.0          BiocFileCache_1.14.0         
  [3] igraph_1.2.6                  lazyeval_0.2.2               
  [5] proxyC_0.1.5                  splines_4.0.3                
  [7] BiocParallel_1.24.1           digest_0.6.27                
  [9] htmltools_0.5.0               viridis_0.5.1                
 [11] magrittr_2.0.1                memoise_1.1.0                
 [13] cluster_2.1.0                 mixtools_1.2.0               
 [15] limma_3.46.0                  Biostrings_2.58.0            
 [17] graphlayouts_0.7.1            RcppParallel_5.0.2           
 [19] askpass_1.1                   prettyunits_1.1.1            
 [21] colorspace_2.0-0              blob_1.2.1                   
 [23] rappdirs_0.3.1                ggrepel_0.8.2                
 [25] xfun_0.19                     dplyr_1.0.2                  
 [27] crayon_1.3.4                  RCurl_1.98-1.2               
 [29] survival_3.2-7                glue_1.4.2                   
 [31] polyclip_1.10-0               gtable_0.3.0                 
 [33] zlibbioc_1.36.0               XVector_0.30.0               
 [35] BiocSingular_1.6.0            kernlab_0.9-29               
 [37] Rhdf5lib_1.12.0               scales_1.1.1                 
 [39] edgeR_3.32.0                  DBI_1.1.0                    
 [41] Rcpp_1.0.5                    viridisLite_0.3.0            
 [43] xtable_1.8-4                  progress_1.2.2               
 [45] dqrng_0.2.1                   bit_4.0.4                    
 [47] rsvd_1.0.3                    proxy_0.4-24                 
 [49] httr_1.4.2                    ellipsis_0.3.1               
 [51] pkgconfig_2.0.3               XML_3.99-0.5                 
 [53] farver_2.0.3                  dbplyr_2.0.0                 
 [55] locfit_1.5-9.4                labeling_0.4.2               
 [57] tidyselect_1.1.0              rlang_0.4.9                  
 [59] later_1.1.0.1                 munsell_0.5.0                
 [61] BiocVersion_3.12.0            tools_4.0.3                  
 [63] generics_0.1.0                RSQLite_2.2.1                
 [65] ExperimentHub_1.16.0          evaluate_0.14                
 [67] stringr_1.4.0                 fastmap_1.0.1                
 [69] yaml_2.2.1                    knitr_1.30                   
 [71] bit64_4.0.5                   fs_1.5.0                     
 [73] tidygraph_1.2.0               purrr_0.3.4                  
 [75] ggraph_2.0.4                  nlme_3.1-150                 
 [77] sparseMatrixStats_1.2.0       whisker_0.4                  
 [79] mime_0.9                      xml2_1.3.2                   
 [81] biomaRt_2.46.0                compiler_4.0.3               
 [83] rstudioapi_0.13               beeswarm_0.2.3               
 [85] curl_4.3                      interactiveDisplayBase_1.28.0
 [87] tibble_3.0.4                  statmod_1.4.35               
 [89] tweenr_1.0.1                  stringi_1.5.3                
 [91] bluster_1.0.0                 lattice_0.20-41              
 [93] hopach_2.50.0                 ProtGenerics_1.22.0          
 [95] vctrs_0.3.5                   rhdf5filters_1.2.0           
 [97] pillar_1.4.7                  lifecycle_0.2.0              
 [99] BiocManager_1.30.10           BiocNeighbors_1.8.2          
[101] bitops_1.0-6                  irlba_2.3.3                  
[103] httpuv_1.5.4                  rtracklayer_1.50.0           
[105] R6_2.5.0                      promises_1.1.1               
[107] gridExtra_2.3                 vipor_0.4.5                  
[109] MASS_7.3-53                   assertthat_0.2.1             
[111] openssl_1.4.3                 rprojroot_2.0.2              
[113] minpack.lm_1.2-1              withr_2.3.0                  
[115] GenomicAlignments_1.26.0      Rsamtools_2.6.0              
[117] GenomeInfoDbData_1.2.4        diptest_0.75-7               
[119] mgcv_1.8-33                   hms_0.5.3                    
[121] grid_4.0.3                    beachmat_2.6.2               
[123] tidyr_1.1.2                   rmarkdown_2.5                
[125] DelayedMatrixStats_1.12.1     segmented_1.3-0              
[127] git2r_0.27.1                  ggforce_0.3.2                
[129] shiny_1.5.0                   ggbeeswarm_0.6.0             </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
