<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Seurat</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span. { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">project-cell-type-annotation</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Preprocessing
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="dataprep_baron.html">Baron Dataset</a>
    </li>
    <li>
      <a href="dataprep_muraro.html">Muraro Dataset</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Methods
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="SingleR.html">SingleR</a>
    </li>
    <li>
      <a href="scClassify.html">scClassify</a>
    </li>
    <li>
      <a href="Seurat.html">Seurat</a>
    </li>
  </ul>
</li>
<li>
  <a href="evaluation.html">Evaluation</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Seurat</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-01-07
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>project-cell-type-annotation/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsta426hs2020projectcelltypeannotationtree7d96421cc0c4aa2d2d4ec8a5a4d9cea7d21de80ftargetblank7d96421a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sta426hs2020/project-cell-type-annotation/tree/7d96421cc0c4aa2d2d4ec8a5a4d9cea7d21de80f" target="_blank">7d96421</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsta426hs2020projectcelltypeannotationtree7d96421cc0c4aa2d2d4ec8a5a4d9cea7d21de80ftargetblank7d96421a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sta426hs2020/project-cell-type-annotation/tree/7d96421cc0c4aa2d2d4ec8a5a4d9cea7d21de80f" target="_blank">7d96421</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Unstaged changes:
    Modified:   analysis/evaluation.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Seurat.Rmd</code>) and HTML (<code>docs/Seurat.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/7d96421cc0c4aa2d2d4ec8a5a4d9cea7d21de80f/analysis/Seurat.Rmd" target="_blank">7d96421</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-07
</td>
<td>
wflow_publish(“analysis/Seurat.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/777844c4cf6d4c2cc2090dd9aec9e4e4585b377b/docs/Seurat.html" target="_blank">777844c</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/f969a51f72c687bf62994cbb640b8d593c10c272/analysis/Seurat.Rmd" target="_blank">f969a51</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-06
</td>
<td>
add everything
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/b744d3b085bcc1778af99b49f21c3669d189dfde/docs/Seurat.html" target="_blank">b744d3b</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/2ec2e351c81888254dc19b965f9c5260fe9782df/analysis/Seurat.Rmd" target="_blank">2ec2e35</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-06
</td>
<td>
add stines comments to theory part
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/f84db0a70a5ee188d9dcf46d88d7e706baaf0b1e/docs/Seurat.html" target="_blank">f84db0a</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/a8509f727c3f8b6e6da544a10b1e844b4524b850/analysis/Seurat.Rmd" target="_blank">a8509f7</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-06
</td>
<td>
add felix’ changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/2f540fdd78351de014541b8ad61d151fd2b2425d/docs/Seurat.html" target="_blank">2f540fd</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/200299eb549d4390b6a996f77edb2d7cc9b3b875/analysis/Seurat.Rmd" target="_blank">200299e</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-04
</td>
<td>
add some comments
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/20d4320007b8571640d144e4faf63118d659509f/docs/Seurat.html" target="_blank">20d4320</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/5716e2480a1144996a2a92401ef05583e3cbc3f5/analysis/Seurat.Rmd" target="_blank">5716e24</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-04
</td>
<td>
Seurat Theory Text
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/a675d967be6735704473586e08c96301163abf72/docs/Seurat.html" target="_blank">a675d96</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/e757d05c5e580d12c682384314de8dcddcc03a80/analysis/Seurat.Rmd" target="_blank">e757d05</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-03
</td>
<td>
updated Seurat file
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/6e02fb204a1e1165bd2337a0814bfe4d40f1dfd0/docs/Seurat.html" target="_blank">6e02fb2</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/72ee1b5b7bc9a609cbec948e22963ccf059b01d5/analysis/Seurat.Rmd" target="_blank">72ee1b5</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-02
</td>
<td>
adjust graphs and export predictions
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/68ec84461aea3fc83c30469f055551ddc7c07bf8/docs/Seurat.html" target="_blank">68ec844</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/872c015548cafbc629456f5aeeaed6b9a20a3476/analysis/Seurat.Rmd" target="_blank">872c015</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-30
</td>
<td>
verbose = FALSE
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/55af0ea77e86f76336afd2fd9ad221536d25cb8e/docs/Seurat.html" target="_blank">55af0ea</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/1ae07f41d7af0f3b08c77dcf57dbd45e848b9956/analysis/Seurat.Rmd" target="_blank">1ae07f4</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-30
</td>
<td>
no warnings
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/d2ceb8b3e4f108618e57a4b9ba18fb61f8d825ce/docs/Seurat.html" target="_blank">d2ceb8b</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/f80d9f6f1a735e39f18cebf266b6b38f4d5d35bf/analysis/Seurat.Rmd" target="_blank">f80d9f6</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-30
</td>
<td>
Seurat predictions
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="theory" class="section level1">
<h1>Theory</h1>
<p>The R package <em>Seurat</em> is a tool for quality control, analysis, and exploration of single-cell RNA-seq data. The version <em>Seurat v3</em> introduced an anchoring procedure that is designed to integrate across technologies and modalities. In short, the method determines anchors between a pair of dataset and then uses these anchors to transfer cell type labels from the reference dataset to the query dataset. <span class="citation">(Stuart et al. 2019)</span></p>
<p>Method description:</p>
<ul>
<li><p>After normalization, feature selection and dimensionality reduction, the key step of <em>Seurat</em> is the unsupervised determination of anchors between two datasets. An anchor represents two cells (one from each dataset), which are predicted to come from an identical biological state.</p></li>
<li><p>Anchors are found in the low-dimensional space, but scored in the high-dimensional space of the expression data in a procedure further described in <span class="citation">Stuart et al. (2019)</span>. Anchors are then weighted based on their score and the distance between the query cell and the anchor and we obtain a weight Matrix <span class="math inline">\(M\)</span>, which represents the association between all pairs of query cells and anchors.</p></li>
<li><p>After having found, scored and weighted the anchors, we can use them to transfer metadata (in our case cell labels) from the reference to the query dataset. The predictions <span class="math inline">\(P_l\)</span> are calculated by multiplying the anchor classification matrix <span class="math inline">\(L\)</span> with the weights matrix <span class="math inline">\(M\)</span>: <span class="math display">\[P_l=LW^T\]</span> The binary classification matrix <span class="math inline">\(L\)</span> contains the classification information for each anchor cell in the reference dataset</p></li>
</ul>
<p>The principle of anchors can also be used to first integrate multiple reference datasets into a single reference and then in a second step transfer cell types from combined references onto a query dataset. A further posibility of <em>Seurat</em> is to integrate data from different modalities (e.g. integrate scRNA-seq data with scATAC-seq data) into a single dataset, which might resolve differences between cell types that would remain unresolvable with either scRNA-seq or scATAC-seq data alone.</p>
<p>The workflow below is adopted from the <a href="https://satijalab.org/seurat/v4.0/reference_mapping.html">official webpage of the software</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(SingleCellExperiment))</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(scRNAseq))</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(AnnotationHub))</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(Seurat))</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(scater))</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(cowplot))</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(uwot))</a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(ggplot2))</a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(patchwork))</a></code></pre></div>
</div>
<div id="loading-the-data" class="section level1">
<h1>Loading the data</h1>
<p>We read in the datasets:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># muraro</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">muraro &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;data/sce_muraro.RDS&quot;</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">true_labels_muraro &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">table</span>(muraro<span class="op">$</span>label))</a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="kw">names</span>(true_labels_muraro) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;cell_type&quot;</span>, <span class="st">&quot;True Muraro&quot;</span>)</a>
<a class="sourceLine" id="cb2-5" data-line-number="5"></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="co"># baron</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7">baron &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;data/sce_baron.RDS&quot;</span>)</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">true_labels_baron &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">table</span>(baron<span class="op">$</span>label))</a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="kw">names</span>(true_labels_baron) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;cell_type&quot;</span>, <span class="st">&quot;True Baron&quot;</span>)</a></code></pre></div>
<p>Next, we have to convert the datasets to <em>Seurat</em> objects. We are preprocessing the data using <em>SCTransform()</em>, which normalizes, scales and finds variable features for the data. The function uses regularized negative binomial regression to normalize and stabilize the variance <span class="citation">(Hafemeister and Satija 2019)</span>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">muraro &lt;-<span class="st"> </span><span class="kw">as.Seurat</span>(muraro)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">baron &lt;-<span class="st"> </span><span class="kw">as.Seurat</span>(baron)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"></a>
<a class="sourceLine" id="cb3-4" data-line-number="4">baron &lt;-<span class="st"> </span><span class="kw">SCTransform</span>(baron, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">baron &lt;-<span class="st"> </span><span class="kw">RunPCA</span>(baron, <span class="dt">npcs =</span> <span class="dv">30</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb3-6" data-line-number="6">baron &lt;-<span class="st"> </span><span class="kw">RunUMAP</span>(baron, <span class="dt">reduction =</span> <span class="st">&quot;pca&quot;</span>, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">30</span>)</a>
<a class="sourceLine" id="cb3-7" data-line-number="7">baron &lt;-<span class="st"> </span><span class="kw">RunTSNE</span>(baron, <span class="dt">reduction =</span> <span class="st">&quot;pca&quot;</span>, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">30</span>)</a>
<a class="sourceLine" id="cb3-8" data-line-number="8"></a>
<a class="sourceLine" id="cb3-9" data-line-number="9">muraro &lt;-<span class="st"> </span><span class="kw">SCTransform</span>(muraro, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb3-10" data-line-number="10">muraro &lt;-<span class="st"> </span><span class="kw">RunPCA</span>(muraro, <span class="dt">npcs =</span> <span class="dv">30</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb3-11" data-line-number="11">muraro &lt;-<span class="st"> </span><span class="kw">RunUMAP</span>(muraro, <span class="dt">reduction =</span> <span class="st">&quot;pca&quot;</span>, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">30</span>)</a>
<a class="sourceLine" id="cb3-12" data-line-number="12">muraro &lt;-<span class="st"> </span><span class="kw">RunTSNE</span>(muraro, <span class="dt">reduction =</span> <span class="st">&quot;pca&quot;</span>, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">30</span>)</a></code></pre></div>
</div>
<div id="annotating-baron-2016" class="section level1">
<h1>Annotating Baron (2016)</h1>
<p>We start with Baron (2016) as our query dataset and Muraro (2016) as the reference.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">anchors &lt;-<span class="st"> </span><span class="kw">FindTransferAnchors</span>(</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">  <span class="dt">reference =</span> muraro,</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">  <span class="dt">query =</span> baron,</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">  <span class="dt">normalization.method =</span> <span class="st">&quot;SCT&quot;</span>,</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">  <span class="dt">reference.reduction =</span> <span class="st">&quot;pca&quot;</span>,</a>
<a class="sourceLine" id="cb4-6" data-line-number="6">  <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">30</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7">)</a>
<a class="sourceLine" id="cb4-8" data-line-number="8"></a>
<a class="sourceLine" id="cb4-9" data-line-number="9">predictions &lt;-<span class="st"> </span><span class="kw">TransferData</span>(<span class="dt">anchorset =</span> anchors, <span class="dt">refdata =</span> muraro<span class="op">$</span>label, </a>
<a class="sourceLine" id="cb4-10" data-line-number="10">    <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">30</span>)</a>
<a class="sourceLine" id="cb4-11" data-line-number="11"></a>
<a class="sourceLine" id="cb4-12" data-line-number="12">baron &lt;-<span class="st"> </span><span class="kw">AddMetaData</span>(baron, <span class="dt">metadata =</span> predictions)</a></code></pre></div>
<div id="predictions" class="section level2">
<h2>Predictions</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">pred_labels_baron &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">table</span>(baron<span class="op">$</span>predicted.id))</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw">names</span>(pred_labels_baron) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;cell_type&quot;</span>, <span class="st">&quot;Predicted by Seurat&quot;</span>)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">labels_baron_Seurat &lt;-<span class="st"> </span><span class="kw">merge</span>(pred_labels_baron, true_labels_baron)</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">knitr<span class="op">::</span><span class="kw">kable</span>(labels_baron_Seurat) <span class="co">#distribution of predicted labels</span></a></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">cell_type</th>
<th align="right">Predicted by Seurat</th>
<th align="right">True Baron</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">acinar</td>
<td align="right">1147</td>
<td align="right">958</td>
</tr>
<tr class="even">
<td align="left">alpha</td>
<td align="right">2329</td>
<td align="right">2326</td>
</tr>
<tr class="odd">
<td align="left">beta</td>
<td align="right">2523</td>
<td align="right">2525</td>
</tr>
<tr class="even">
<td align="left">delta</td>
<td align="right">611</td>
<td align="right">601</td>
</tr>
<tr class="odd">
<td align="left">ductal</td>
<td align="right">892</td>
<td align="right">1077</td>
</tr>
<tr class="even">
<td align="left">endothelial</td>
<td align="right">241</td>
<td align="right">252</td>
</tr>
<tr class="odd">
<td align="left">epsilon</td>
<td align="right">13</td>
<td align="right">18</td>
</tr>
<tr class="even">
<td align="left">gamma</td>
<td align="right">256</td>
<td align="right">255</td>
</tr>
</tbody>
</table>
</div>
<div id="t-sne-plots" class="section level2">
<h2>t-SNE plots</h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">p1<span class="fl">.1</span> =<span class="st"> </span><span class="kw">DimPlot</span>(baron, <span class="dt">reduction =</span> <span class="st">&quot;tsne&quot;</span>, <span class="dt">group.by =</span> <span class="st">&quot;predicted.id&quot;</span>, <span class="dt">label =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="dt">label =</span> <span class="st">&quot;Predicted Cell Type&quot;</span>)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"></a>
<a class="sourceLine" id="cb6-3" data-line-number="3">p1<span class="fl">.2</span> =<span class="st"> </span><span class="kw">DimPlot</span>(baron, <span class="dt">reduction =</span> <span class="st">&quot;tsne&quot;</span>, <span class="dt">group.by =</span> <span class="st">&quot;label&quot;</span>, <span class="dt">label =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st">  </span><span class="kw">ggtitle</span>(<span class="dt">label =</span> <span class="st">&quot;True Cell Type&quot;</span>)</a>
<a class="sourceLine" id="cb6-4" data-line-number="4"></a>
<a class="sourceLine" id="cb6-5" data-line-number="5">ggpubr<span class="op">::</span><span class="kw">ggarrange</span>(p1<span class="fl">.1</span>, p1<span class="fl">.2</span>, <span class="dt">common.legend =</span> <span class="ot">TRUE</span>, <span class="dt">legend =</span> <span class="st">&quot;bottom&quot;</span>)</a></code></pre></div>
<p><img src="figure/Seurat.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/f84db0a70a5ee188d9dcf46d88d7e706baaf0b1e/docs/figure/Seurat.Rmd/unnamed-chunk-1-1.png" target="_blank">f84db0a</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Overall Seurat’s predictions for the Baron dataset seem to be quite accurate. However, we can see in the t-SNE plots that there is one minor cluster of ductal cells in the center right was misclassified as acinar cells. This is also reflected in the table above: Seurat classified to many cells as acinar and too few as ductal.</p>
</div>
</div>
<div id="annotating-muraro-2016" class="section level1">
<h1>Annotating Muraro (2016)</h1>
<p>Now we switch reference and query dataset and perform the same analysis.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">anchors &lt;-<span class="st"> </span><span class="kw">FindTransferAnchors</span>(</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">  <span class="dt">reference =</span> baron,</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">  <span class="dt">query =</span> muraro,</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">  <span class="dt">normalization.method =</span> <span class="st">&quot;SCT&quot;</span>,</a>
<a class="sourceLine" id="cb7-5" data-line-number="5">  <span class="dt">reference.reduction =</span> <span class="st">&quot;pca&quot;</span>,</a>
<a class="sourceLine" id="cb7-6" data-line-number="6">  <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">30</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7">)</a>
<a class="sourceLine" id="cb7-8" data-line-number="8"></a>
<a class="sourceLine" id="cb7-9" data-line-number="9">predictions &lt;-<span class="st"> </span><span class="kw">TransferData</span>(<span class="dt">anchorset =</span> anchors, <span class="dt">refdata =</span> baron<span class="op">$</span>label, </a>
<a class="sourceLine" id="cb7-10" data-line-number="10">    <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">30</span>)</a>
<a class="sourceLine" id="cb7-11" data-line-number="11"></a>
<a class="sourceLine" id="cb7-12" data-line-number="12">muraro &lt;-<span class="st"> </span><span class="kw">AddMetaData</span>(muraro, <span class="dt">metadata =</span> predictions)</a></code></pre></div>
<div id="predictions-1" class="section level2">
<h2>Predictions</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">pred_labels_muraro &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">table</span>(muraro<span class="op">$</span>predicted.id))</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw">names</span>(pred_labels_muraro) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;cell_type&quot;</span>, <span class="st">&quot;Predicted by Seurat&quot;</span>)</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">labels_muraro_Seurat &lt;-<span class="st"> </span><span class="kw">merge</span>(pred_labels_muraro, true_labels_muraro)</a>
<a class="sourceLine" id="cb8-4" data-line-number="4">knitr<span class="op">::</span><span class="kw">kable</span>(labels_muraro_Seurat) <span class="co">#distribution of predicted labels</span></a></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">cell_type</th>
<th align="right">Predicted by Seurat</th>
<th align="right">True Muraro</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">acinar</td>
<td align="right">53</td>
<td align="right">218</td>
</tr>
<tr class="even">
<td align="left">alpha</td>
<td align="right">778</td>
<td align="right">803</td>
</tr>
<tr class="odd">
<td align="left">beta</td>
<td align="right">447</td>
<td align="right">446</td>
</tr>
<tr class="even">
<td align="left">delta</td>
<td align="right">188</td>
<td align="right">191</td>
</tr>
<tr class="odd">
<td align="left">ductal</td>
<td align="right">406</td>
<td align="right">242</td>
</tr>
<tr class="even">
<td align="left">endothelial</td>
<td align="right">42</td>
<td align="right">20</td>
</tr>
<tr class="odd">
<td align="left">epsilon</td>
<td align="right">3</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">gamma</td>
<td align="right">105</td>
<td align="right">99</td>
</tr>
</tbody>
</table>
</div>
<div id="t-sne-plots-1" class="section level2">
<h2>t-SNE plots</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">p2<span class="fl">.1</span> =<span class="st"> </span><span class="kw">DimPlot</span>(muraro, <span class="dt">reduction =</span> <span class="st">&quot;tsne&quot;</span>, <span class="dt">group.by =</span> <span class="st">&quot;predicted.id&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="dt">label =</span> <span class="st">&quot;Predicted Cell Type&quot;</span>)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"></a>
<a class="sourceLine" id="cb9-3" data-line-number="3">p2<span class="fl">.2</span> =<span class="st"> </span><span class="kw">DimPlot</span>(muraro, <span class="dt">reduction =</span> <span class="st">&quot;tsne&quot;</span>, <span class="dt">group.by =</span> <span class="st">&quot;label&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="dt">label =</span> <span class="st">&quot;True Cell Type&quot;</span>)</a>
<a class="sourceLine" id="cb9-4" data-line-number="4"></a>
<a class="sourceLine" id="cb9-5" data-line-number="5">ggpubr<span class="op">::</span><span class="kw">ggarrange</span>(p2<span class="fl">.1</span>, p2<span class="fl">.2</span>, <span class="dt">common.legend =</span> <span class="ot">TRUE</span>, <span class="dt">legend =</span> <span class="st">&quot;bottom&quot;</span>)</a></code></pre></div>
<p><img src="figure/Seurat.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/f84db0a70a5ee188d9dcf46d88d7e706baaf0b1e/docs/figure/Seurat.Rmd/unnamed-chunk-2-1.png" target="_blank">f84db0a</a>
</td>
<td>
speckerf
</td>
<td>
2021-01-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Overall Seurat’s predictions for the Muraro dataset seem to be quite accurate with one exception. The t-SNE plots shows that the a large part of the cluster in the top was misclassified as ductal cells, where they are supposed to be acinar cells. In this case, the same two cell types (acinar and ductal) caused troubles,</p>
<p>This is also reflected in the table above: Seurat classified to many cells as acinar and too few as ductal.</p>
</div>
</div>
<div id="saving-objects-for-evaluation" class="section level1">
<h1>Saving objects for evaluation</h1>
<p>In order to compare the predictions by the three methods, we again remove cells that where not fully assigned to a label by <em>scClassify</em>. This is done, because the unassigned cells are presumably harder to correctly classify for every method and therefore we can only compare the three methods, if these cells are exluded in the evaluation.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="co">#extract true and predicted labels</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2">pred.baron.labels &lt;-<span class="st"> </span>baron<span class="op">$</span>predicted.id</a>
<a class="sourceLine" id="cb10-3" data-line-number="3">true.baron.labels &lt;-<span class="st"> </span>baron<span class="op">$</span>label</a>
<a class="sourceLine" id="cb10-4" data-line-number="4"></a>
<a class="sourceLine" id="cb10-5" data-line-number="5">unassigned_baron &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file=</span><span class="st">&quot;data/baron_unassigned_scClassify.rds&quot;</span>)</a>
<a class="sourceLine" id="cb10-6" data-line-number="6"></a>
<a class="sourceLine" id="cb10-7" data-line-number="7">pred.baron.labels &lt;-<span class="st"> </span>pred.baron.labels[<span class="op">!</span>unassigned_baron]</a>
<a class="sourceLine" id="cb10-8" data-line-number="8"><span class="kw">saveRDS</span>(pred.baron.labels, <span class="dt">file=</span><span class="st">&quot;data/Seurat.pred.baron.labels.rds&quot;</span>)</a>
<a class="sourceLine" id="cb10-9" data-line-number="9"></a>
<a class="sourceLine" id="cb10-10" data-line-number="10">true.baron.labels &lt;-<span class="st"> </span>true.baron.labels[<span class="op">!</span>unassigned_baron]</a>
<a class="sourceLine" id="cb10-11" data-line-number="11"><span class="kw">saveRDS</span>(true.baron.labels, <span class="dt">file=</span><span class="st">&quot;data/Seurat.true.baron.labels.rds&quot;</span>)</a>
<a class="sourceLine" id="cb10-12" data-line-number="12"></a>
<a class="sourceLine" id="cb10-13" data-line-number="13">pred.muraro.labels &lt;-<span class="st"> </span>muraro<span class="op">$</span>predicted.id</a>
<a class="sourceLine" id="cb10-14" data-line-number="14">true.muraro.labels &lt;-<span class="st"> </span>muraro<span class="op">$</span>label</a>
<a class="sourceLine" id="cb10-15" data-line-number="15"></a>
<a class="sourceLine" id="cb10-16" data-line-number="16">unassigned_muraro &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file=</span><span class="st">&quot;data/muraro_unassigned_scClassify.rds&quot;</span>)</a>
<a class="sourceLine" id="cb10-17" data-line-number="17"></a>
<a class="sourceLine" id="cb10-18" data-line-number="18">pred.muraro.labels &lt;-<span class="st"> </span>pred.muraro.labels[<span class="op">!</span>unassigned_muraro]</a>
<a class="sourceLine" id="cb10-19" data-line-number="19"><span class="kw">saveRDS</span>(pred.muraro.labels, <span class="dt">file=</span><span class="st">&quot;data/Seurat.pred.muraro.labels.rds&quot;</span>)</a>
<a class="sourceLine" id="cb10-20" data-line-number="20"></a>
<a class="sourceLine" id="cb10-21" data-line-number="21">true.muraro.labels &lt;-<span class="st"> </span>true.muraro.labels[<span class="op">!</span>unassigned_muraro]</a>
<a class="sourceLine" id="cb10-22" data-line-number="22"><span class="kw">saveRDS</span>(true.muraro.labels, <span class="dt">file=</span><span class="st">&quot;data/Seurat.true.muraro.labels.rds&quot;</span>)</a></code></pre></div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs">
<div id="ref-hafemeister2019normalization">
<p>Hafemeister, Christoph, and Rahul Satija. 2019. “Normalization and Variance Stabilization of Single-Cell Rna-Seq Data Using Regularized Negative Binomial Regression.” <em>Genome Biology</em> 20 (1). Springer: 1–15.</p>
</div>
<div id="ref-stuart2019comprehensive">
<p>Stuart, Tim, Andrew Butler, Paul Hoffman, Christoph Hafemeister, Efthymia Papalexi, William M Mauck III, Yuhan Hao, Marlon Stoeckius, Peter Smibert, and Rahul Satija. 2019. “Comprehensive Integration of Single-Cell Data.” <em>Cell</em> 177 (7). Elsevier: 1888–1902.</p>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] de_CH.UTF-8/de_CH.UTF-8/de_CH.UTF-8/C/de_CH.UTF-8/de_CH.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] patchwork_1.1.1             uwot_0.1.10                
 [3] Matrix_1.2-18               cowplot_1.1.0              
 [5] scater_1.18.3               ggplot2_3.3.2              
 [7] Seurat_3.9.9.9024           AnnotationHub_2.22.0       
 [9] BiocFileCache_1.14.0        dbplyr_2.0.0               
[11] scRNAseq_2.4.0              SingleCellExperiment_1.12.0
[13] SummarizedExperiment_1.20.0 Biobase_2.50.0             
[15] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
[17] IRanges_2.24.0              S4Vectors_0.28.0           
[19] BiocGenerics_0.36.0         MatrixGenerics_1.2.0       
[21] matrixStats_0.57.0          workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] reticulate_1.18               tidyselect_1.1.0             
  [3] RSQLite_2.2.1                 AnnotationDbi_1.52.0         
  [5] htmlwidgets_1.5.3             grid_4.0.3                   
  [7] BiocParallel_1.24.1           Rtsne_0.15                   
  [9] munsell_0.5.0                 codetools_0.2-18             
 [11] ica_1.0-2                     future_1.21.0                
 [13] miniUI_0.1.1.1                withr_2.3.0                  
 [15] colorspace_2.0-0              highr_0.8                    
 [17] knitr_1.30                    rstudioapi_0.13              
 [19] ROCR_1.0-11                   ggsignif_0.6.0               
 [21] tensor_1.5                    listenv_0.8.0                
 [23] labeling_0.4.2                git2r_0.27.1                 
 [25] GenomeInfoDbData_1.2.4        polyclip_1.10-0              
 [27] farver_2.0.3                  bit64_4.0.5                  
 [29] rprojroot_2.0.2               parallelly_1.22.0            
 [31] vctrs_0.3.6                   generics_0.1.0               
 [33] xfun_0.19                     R6_2.5.0                     
 [35] ggbeeswarm_0.6.0              rsvd_1.0.3                   
 [37] AnnotationFilter_1.14.0       bitops_1.0-6                 
 [39] spatstat.utils_1.17-0         DelayedArray_0.16.0          
 [41] assertthat_0.2.1              promises_1.1.1               
 [43] scales_1.1.1                  beeswarm_0.2.3               
 [45] gtable_0.3.0                  beachmat_2.6.2               
 [47] globals_0.14.0                goftest_1.2-2                
 [49] ensembldb_2.14.0              rlang_0.4.9                  
 [51] splines_4.0.3                 rstatix_0.6.0                
 [53] rtracklayer_1.50.0            lazyeval_0.2.2               
 [55] broom_0.7.2                   BiocManager_1.30.10          
 [57] yaml_2.2.1                    reshape2_1.4.4               
 [59] abind_1.4-5                   backports_1.2.1              
 [61] GenomicFeatures_1.42.1        httpuv_1.5.4                 
 [63] tools_4.0.3                   ellipsis_0.3.1               
 [65] RColorBrewer_1.1-2            ggridges_0.5.2               
 [67] Rcpp_1.0.5                    plyr_1.8.6                   
 [69] sparseMatrixStats_1.2.0       progress_1.2.2               
 [71] zlibbioc_1.36.0               purrr_0.3.4                  
 [73] RCurl_1.98-1.2                prettyunits_1.1.1            
 [75] ggpubr_0.4.0                  rpart_4.1-15                 
 [77] openssl_1.4.3                 deldir_0.2-3                 
 [79] viridis_0.5.1                 pbapply_1.4-3                
 [81] zoo_1.8-8                     haven_2.3.1                  
 [83] ggrepel_0.9.0                 cluster_2.1.0                
 [85] fs_1.5.0                      magrittr_2.0.1               
 [87] RSpectra_0.16-0               data.table_1.13.4            
 [89] scattermore_0.7               openxlsx_4.2.3               
 [91] lmtest_0.9-38                 RANN_2.6.1                   
 [93] whisker_0.4                   ProtGenerics_1.22.0          
 [95] fitdistrplus_1.1-3            hms_0.5.3                    
 [97] mime_0.9                      evaluate_0.14                
 [99] xtable_1.8-4                  XML_3.99-0.5                 
[101] rio_0.5.16                    readxl_1.3.1                 
[103] gridExtra_2.3                 compiler_4.0.3               
[105] biomaRt_2.46.0                tibble_3.0.4                 
[107] KernSmooth_2.23-18            crayon_1.3.4                 
[109] htmltools_0.5.0               mgcv_1.8-33                  
[111] later_1.1.0.1                 tidyr_1.1.2                  
[113] DBI_1.1.0                     ExperimentHub_1.16.0         
[115] MASS_7.3-53                   rappdirs_0.3.1               
[117] car_3.0-10                    igraph_1.2.6                 
[119] forcats_0.5.0                 pkgconfig_2.0.3              
[121] GenomicAlignments_1.26.0      foreign_0.8-80               
[123] plotly_4.9.2.2                scuttle_1.0.3                
[125] xml2_1.3.2                    vipor_0.4.5                  
[127] XVector_0.30.0                stringr_1.4.0                
[129] digest_0.6.27                 sctransform_0.3.2            
[131] RcppAnnoy_0.0.18              spatstat.data_1.7-0          
[133] Biostrings_2.58.0             cellranger_1.1.0             
[135] rmarkdown_2.5                 leiden_0.3.6                 
[137] DelayedMatrixStats_1.12.1     curl_4.3                     
[139] shiny_1.5.0                   Rsamtools_2.6.0              
[141] lifecycle_0.2.0               nlme_3.1-150                 
[143] jsonlite_1.7.2                carData_3.0-4                
[145] BiocNeighbors_1.8.2           viridisLite_0.3.0            
[147] askpass_1.1                   pillar_1.4.7                 
[149] lattice_0.20-41               fastmap_1.0.1                
[151] httr_1.4.2                    survival_3.2-7               
[153] interactiveDisplayBase_1.28.0 glue_1.4.2                   
[155] zip_2.1.1                     spatstat_1.64-1              
[157] png_0.1-7                     BiocVersion_3.12.0           
[159] bit_4.0.4                     stringi_1.5.3                
[161] blob_1.2.1                    BiocSingular_1.6.0           
[163] memoise_1.1.0                 dplyr_1.0.2                  
[165] irlba_2.3.3                   future.apply_1.6.0           </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
