<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>SingleR</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span. { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">project-cell-type-annotation</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Preprocessing
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="dataprep_baron.html">Baron Dataset</a>
    </li>
    <li>
      <a href="dataprep_muraro.html">Muraro Dataset</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Methods
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="SingleR.html">SingleR</a>
    </li>
    <li>
      <a href="scClassify.html">scClassify</a>
    </li>
    <li>
      <a href="Seurat.html">Seurat</a>
    </li>
  </ul>
</li>
<li>
  <a href="evaluation.html">Evaluation</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">SingleR</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-01-07
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>project-cell-type-annotation/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsta426hs2020projectcelltypeannotationtreed77249e276ea18019baeae6c2de0f613ac5c3407targetblankd77249ea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sta426hs2020/project-cell-type-annotation/tree/d77249e276ea18019baeae6c2de0f613ac5c3407" target="_blank">d77249e</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsta426hs2020projectcelltypeannotationtreed77249e276ea18019baeae6c2de0f613ac5c3407targetblankd77249ea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sta426hs2020/project-cell-type-annotation/tree/d77249e276ea18019baeae6c2de0f613ac5c3407" target="_blank">d77249e</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    analysis/site_libs/

Unstaged changes:
    Modified:   analysis/evaluation.Rmd
    Modified:   analysis/scClassify.Rmd
    Modified:   ref.bibtex

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/SingleR.Rmd</code>) and HTML (<code>docs/SingleR.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/f969a51f72c687bf62994cbb640b8d593c10c272/analysis/SingleR.Rmd" target="_blank">f969a51</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-06
</td>
<td>
add everything
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/a3e15a0a1d425a13547e8c33cf8864350f2c368e/docs/SingleR.html" target="_blank">a3e15a0</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/d545d439e80ab179af96b62e0c47055cc87a60d3/analysis/SingleR.Rmd" target="_blank">d545d43</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-06
</td>
<td>
newest changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/d545d439e80ab179af96b62e0c47055cc87a60d3/docs/SingleR.html" target="_blank">d545d43</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-06
</td>
<td>
newest changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/5f05e74a6f75e22d8e5838280cc1d9bb97d35668/docs/SingleR.html" target="_blank">5f05e74</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/a6bda73771c1f644b90ffd74a98b20b154594280/analysis/SingleR.Rmd" target="_blank">a6bda73</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-06
</td>
<td>
wflow_publish(“analysis/SingleR.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/4ce3466c5e1561cd5c6b82f379831263edd82141/docs/SingleR.html" target="_blank">4ce3466</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/e5ba804726c07be2783d2244d798ca30b1f2aabb/analysis/SingleR.Rmd" target="_blank">e5ba804</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-03
</td>
<td>
wflow_publish(“analysis/SingleR.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/91aeadb557381d3c56000933c1da158740921543/docs/SingleR.html" target="_blank">91aeadb</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/e061e565b86165f9615133e32598d8fe1a3382d4/docs/SingleR.html" target="_blank">e061e56</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/7b3b6e37202bdd4ee1122e5a48750b7254a76fa9/analysis/SingleR.Rmd" target="_blank">7b3b6e3</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-30
</td>
<td>
wflow_git_commit(all = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/543a43c4e52e57046078cc48380673460c376972/docs/SingleR.html" target="_blank">543a43c</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/42d96430fd6e5a660f2587f10e94454ff144f61b/analysis/SingleR.Rmd" target="_blank">42d9643</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-30
</td>
<td>
wflow_publish(all = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/42d96430fd6e5a660f2587f10e94454ff144f61b/docs/SingleR.html" target="_blank">42d9643</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-30
</td>
<td>
wflow_publish(all = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/7868051ec79e75cd9847751432527c1beab1c9cb/docs/SingleR.html" target="_blank">7868051</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/53e5d540b2b437c85097e1dc82a661959a3e8464/analysis/SingleR.Rmd" target="_blank">53e5d54</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-28
</td>
<td>
exclude data prep steps
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/89529378cacf0c739e380df824985af967261313/docs/SingleR.html" target="_blank">8952937</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/f5181befb733a5aa4ac183f6d87ba358a29f397c/analysis/SingleR.Rmd" target="_blank">f5181be</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-17
</td>
<td>
wflow_publish(c(“analysis/SingleR.Rmd”, “analysis/SingleR_no_label_adjustment.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/251dc4ad0e992135c7687e4f5b950a3e0f8851f9/docs/SingleR.html" target="_blank">251dc4a</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/bdad24837696bcdcf7b85f9a77eb61ce7a7e6832/analysis/SingleR.Rmd" target="_blank">bdad248</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-16
</td>
<td>
wflow_publish(c(“analysis/SingleR.Rmd”, “analysis/PBMC_datasets.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/2daa9e4d74b12857c4b87a205e90d9b10618b048/docs/SingleR.html" target="_blank">2daa9e4</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/3923c5cb2a75c327637715aab00594e73e4ed7e4/analysis/SingleR.Rmd" target="_blank">3923c5c</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-16
</td>
<td>
modeify yaml header
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sta426hs2020/project-cell-type-annotation/ab3dd46cd2f786b18ab584ab2325f99b1ca41cd0/docs/SingleR.html" target="_blank">ab3dd46</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/6af8188904f1d8c835e71e7e3c28615de2b14505/analysis/SingleR.Rmd" target="_blank">6af8188</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-15
</td>
<td>
wflow_publish(“analysis/SingleR.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/bfdcde440fa10044646375a090c76e587a91d18f/analysis/SingleR.Rmd" target="_blank">bfdcde4</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-15
</td>
<td>
newest changes from tuesday
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="theory" class="section level1">
<h1>Theory</h1>
<p>The package <em>SingleR</em> is an automated annotation method for scRNA-seq data developed in 2019 by Aaron Lun <span class="citation">(Aran et al. 2019)</span>. In short, it’s a robust variant of nearest-neighbors classification. It labels cells in the test dataset based on similarity to the cells in the pre-labelled reference dataset (which can be either bulk RNA-seq or scRNAseq).</p>
<p>Method description <span class="citation">(Lun 2020)</span>:</p>
<ul>
<li><p>Marker genes are identified by performing pairwise comparison between cells from a label of interest and cells from all the other labels. The genes that are upregulated in the label of interest in each pairwise comparison are then identified as marker genes.</p></li>
<li><p>For every cell in the test dataset the Spearman correlation between its expression profile of marker genes and the expression profile of marker genes of each cell in the reference dataset is computed.</p></li>
<li><p>For every test cell a per-label score is then determined as the quantile of the correlations across all (reference) samples with that label.</p></li>
<li><p>The label with the highest score will be SingleR’s prediction for that test cell.</p></li>
</ul>
<p>For this analysis, the <em>SingleR</em> “classic” mode with one reference dataset was used. However, <em>SingleR</em> also has the option of using multiple references, like the other two methods <span class="citation">(Lun 2020)</span>:</p>
<ul>
<li><p>Either performing the annotation with each reference individually and then combining the scores across the references (similar to scClassify)</p></li>
<li><p>Or first creating a combined reference and then performing the annotation with the combined reference (similar to Seurat)</p></li>
</ul>
<p><em>SingleR</em> was run once on the Baron dataset (with Muraro as reference) and vice versa.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(SingleR))</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(scRNAseq))</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(ensembldb))</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(scater))</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(knitr))</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(scuttle))</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(ggpubr))</a></code></pre></div>
</div>
<div id="loading-the-data" class="section level1">
<h1>Loading the data</h1>
<p>The datasets have been preprocessed (see Preprocessing section) and are loaded as SingleCellExperiment objects.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co">#loading muraro dataset</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">muraro &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;data/sce_muraro.rds&quot;</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">true_labels_muraro &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">table</span>(muraro<span class="op">$</span>label))</a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="kw">names</span>(true_labels_muraro) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;cell_type&quot;</span>, <span class="st">&quot;True Muraro&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co">#loading baron dataset</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">baron &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;data/sce_baron.rds&quot;</span>)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">true_labels_baron &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">table</span>(baron<span class="op">$</span>label))</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="kw">names</span>(true_labels_baron) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;cell_type&quot;</span>, <span class="st">&quot;True Baron&quot;</span>)</a></code></pre></div>
</div>
<div id="annotating-baron-2016" class="section level1">
<h1>Annotating Baron (2016)</h1>
<p>Making predictions for the Baron dataset (with the Muraro dataset as reference):</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">pred.baron &lt;-<span class="st"> </span><span class="kw">SingleR</span>(<span class="dt">test =</span> baron, <span class="dt">ref =</span> muraro,</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">    <span class="dt">labels =</span> muraro<span class="op">$</span>label, <span class="dt">de.method =</span> <span class="st">&quot;classic&quot;</span>)</a></code></pre></div>
<div id="predictions" class="section level2">
<h2>Predictions</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">pred_labels_baron &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">table</span>(pred.baron<span class="op">$</span>labels))</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw">names</span>(pred_labels_baron) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;cell_type&quot;</span>, <span class="st">&quot;Predicted by SingleR&quot;</span>)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">labels_baron_singleR &lt;-<span class="st"> </span><span class="kw">merge</span>(pred_labels_baron, true_labels_baron)</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">knitr<span class="op">::</span><span class="kw">kable</span>(labels_baron_singleR, <span class="dt">caption =</span> <span class="st">&quot;Distribution of predicted labels compared to true labels&quot;</span>) </a></code></pre></div>
<table>
<caption>Distribution of predicted labels compared to true labels</caption>
<thead>
<tr class="header">
<th align="left">cell_type</th>
<th align="right">Predicted by SingleR</th>
<th align="right">True Baron</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">acinar</td>
<td align="right">1171</td>
<td align="right">958</td>
</tr>
<tr class="even">
<td align="left">alpha</td>
<td align="right">2332</td>
<td align="right">2326</td>
</tr>
<tr class="odd">
<td align="left">beta</td>
<td align="right">2504</td>
<td align="right">2525</td>
</tr>
<tr class="even">
<td align="left">delta</td>
<td align="right">606</td>
<td align="right">601</td>
</tr>
<tr class="odd">
<td align="left">ductal</td>
<td align="right">888</td>
<td align="right">1077</td>
</tr>
<tr class="even">
<td align="left">endothelial</td>
<td align="right">246</td>
<td align="right">252</td>
</tr>
<tr class="odd">
<td align="left">epsilon</td>
<td align="right">15</td>
<td align="right">18</td>
</tr>
<tr class="even">
<td align="left">gamma</td>
<td align="right">250</td>
<td align="right">255</td>
</tr>
</tbody>
</table>
</div>
<div id="t-sne-plots" class="section level2">
<h2>t-SNE plots</h2>
<p>t-SNE plot of true baron labels and SingleR’s predicted baron labels</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">p1<span class="fl">.1</span> =<span class="st"> </span><span class="kw">plotReducedDim</span>(baron, <span class="st">&quot;TSNE&quot;</span>, <span class="dt">colour_by=</span><span class="st">&quot;label&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;True Cell Type&quot;</span>) <span class="co">#true</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2">p1<span class="fl">.2</span> =<span class="st"> </span><span class="kw">plotReducedDim</span>(baron, <span class="st">&quot;TSNE&quot;</span>, <span class="dt">colour_by=</span><span class="kw">I</span>(pred.baron<span class="op">$</span>labels)) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Predicted Cell Type&quot;</span>)<span class="co">#predicted</span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="kw">ggarrange</span>(p1<span class="fl">.1</span>, p1<span class="fl">.2</span>, <span class="dt">common.legend =</span> <span class="ot">TRUE</span>, <span class="dt">legend =</span> <span class="st">&quot;bottom&quot;</span>)</a></code></pre></div>
<p><img src="figure/SingleR.Rmd/t-SNE plot baron-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-t-SNE-plot-baron-1">
Past versions of “t-SNE plot baron-1.png”
</button>
</p>
<div id="fig-no-spaces-t-SNE-plot-baron-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/5f05e74a6f75e22d8e5838280cc1d9bb97d35668/docs/figure/SingleR.Rmd/t-SNE plot baron-1.png" target="_blank">5f05e74</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/543a43c4e52e57046078cc48380673460c376972/docs/figure/SingleR.Rmd/t-SNE plot baron-1.png" target="_blank">543a43c</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/42d96430fd6e5a660f2587f10e94454ff144f61b/docs/figure/SingleR.Rmd/t-SNE plot baron-1.png" target="_blank">42d9643</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/7868051ec79e75cd9847751432527c1beab1c9cb/docs/figure/SingleR.Rmd/t-SNE plot baron-1.png" target="_blank">7868051</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/89529378cacf0c739e380df824985af967261313/docs/figure/SingleR.Rmd/t-SNE plot baron-1.png" target="_blank">8952937</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Overall SingleR’s prediction for the Baron dataset seems to be quite accurate. However, as can be seen in the t-SNE plots, SingleR seems to have misclassified a cluster of ductal cells as acinar cells in the top middle of the tSNE plot. This is also reflected in the table above: SingleR has classified too many cells as acinar and too little as ductal.</p>
</div>
</div>
<div id="annotating-muraro-2016" class="section level1">
<h1>Annotating Muraro (2016)</h1>
<p>Making predictions for the Muraro dataset (with the Baron dataset as reference):</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">pred.muraro &lt;-<span class="st"> </span><span class="kw">SingleR</span>(<span class="dt">test =</span> muraro, <span class="dt">ref =</span> baron,</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">    <span class="dt">labels =</span> baron<span class="op">$</span>label, <span class="dt">de.method =</span> <span class="st">&quot;classic&quot;</span>)</a></code></pre></div>
<div id="predictions-1" class="section level2">
<h2>Predictions</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">pred_labels_muraro &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">table</span>(pred.muraro<span class="op">$</span>labels))</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw">names</span>(pred_labels_muraro) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;cell_type&quot;</span>, <span class="st">&quot;Predicted by SingleR&quot;</span>)</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">labels_muraro_singleR &lt;-<span class="st"> </span><span class="kw">merge</span>(pred_labels_muraro, true_labels_muraro)</a>
<a class="sourceLine" id="cb8-4" data-line-number="4">knitr<span class="op">::</span><span class="kw">kable</span>(labels_muraro_singleR, <span class="dt">caption =</span> <span class="st">&quot;Distribution of predicted labels compared to true labels&quot;</span>) <span class="co">#distribution of predicted labels</span></a></code></pre></div>
<table>
<caption>Distribution of predicted labels compared to true labels</caption>
<thead>
<tr class="header">
<th align="left">cell_type</th>
<th align="right">Predicted by SingleR</th>
<th align="right">True Muraro</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">acinar</td>
<td align="right">189</td>
<td align="right">218</td>
</tr>
<tr class="even">
<td align="left">alpha</td>
<td align="right">781</td>
<td align="right">803</td>
</tr>
<tr class="odd">
<td align="left">beta</td>
<td align="right">433</td>
<td align="right">446</td>
</tr>
<tr class="even">
<td align="left">delta</td>
<td align="right">204</td>
<td align="right">191</td>
</tr>
<tr class="odd">
<td align="left">ductal</td>
<td align="right">282</td>
<td align="right">242</td>
</tr>
<tr class="even">
<td align="left">endothelial</td>
<td align="right">21</td>
<td align="right">20</td>
</tr>
<tr class="odd">
<td align="left">epsilon</td>
<td align="right">3</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">gamma</td>
<td align="right">109</td>
<td align="right">99</td>
</tr>
</tbody>
</table>
</div>
<div id="t-sne-plots-1" class="section level2">
<h2>t-SNE plots</h2>
<p>t-SNE plot of true muraro labels and SingleR’s predicted muraro labels</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">p2<span class="fl">.1</span> =<span class="st"> </span><span class="kw">plotReducedDim</span>(muraro, <span class="st">&quot;TSNE&quot;</span>, <span class="dt">colour_by=</span><span class="st">&quot;label&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;True Cell Type&quot;</span>) <span class="co">#true</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2">p2<span class="fl">.2</span> =<span class="st"> </span><span class="kw">plotReducedDim</span>(muraro, <span class="st">&quot;TSNE&quot;</span>, <span class="dt">colour_by=</span><span class="kw">I</span>(pred.muraro<span class="op">$</span>labels)) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Predicted Cell Type&quot;</span>)<span class="co">#predicted</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="kw">ggarrange</span>(p2<span class="fl">.1</span>, p2<span class="fl">.2</span>, <span class="dt">common.legend =</span> <span class="ot">TRUE</span>, <span class="dt">legend =</span> <span class="st">&quot;bottom&quot;</span>)</a></code></pre></div>
<p><img src="figure/SingleR.Rmd/t-SNE%20plot%20muraro-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-t-SNE-plot-muraro-1">
Past versions of “t-SNE plot muraro-1.png”
</button>
</p>
<div id="fig-no-spaces-t-SNE-plot-muraro-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/5f05e74a6f75e22d8e5838280cc1d9bb97d35668/docs/figure/SingleR.Rmd/t-SNE plot muraro-1.png" target="_blank">5f05e74</a>
</td>
<td>
stinea98
</td>
<td>
2021-01-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/543a43c4e52e57046078cc48380673460c376972/docs/figure/SingleR.Rmd/t-SNE plot muraro-1.png" target="_blank">543a43c</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/42d96430fd6e5a660f2587f10e94454ff144f61b/docs/figure/SingleR.Rmd/t-SNE plot muraro-1.png" target="_blank">42d9643</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/7868051ec79e75cd9847751432527c1beab1c9cb/docs/figure/SingleR.Rmd/t-SNE plot muraro-1.png" target="_blank">7868051</a>
</td>
<td>
speckerf
</td>
<td>
2020-12-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sta426hs2020/project-cell-type-annotation/blob/89529378cacf0c739e380df824985af967261313/docs/figure/SingleR.Rmd/t-SNE plot muraro-1.png" target="_blank">8952937</a>
</td>
<td>
stinea98
</td>
<td>
2020-12-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Overall SingleR’s prediction for the Muraro dataset seems to be quite accurate. However, as can be seen in the t-SNE plots, SingleR seems to have misclassified a small cluster of acinar cells as ductal cells in the bottom of the tSNE plot. This is also reflected in the table above: SingleR has classified too many cells as ductal and too little as acinar.</p>
</div>
</div>
<div id="saving-objects-for-evaluation" class="section level1">
<h1>Saving objects for evaluation</h1>
<p>The true and predicted label vectors were extracted from the results to be used for the evaluation section later on. Also, cells labelled “unassigned” and thus removed in the scClassify analysis (because they were not assigned to a single cell type, but a parent node) were removed from the SingleR dataset too, in order to be able to compare the methods in the evaluation section.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="co">#Baron: extract true and predicted labels</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2">pred.baron.labels &lt;-<span class="st"> </span>pred.baron<span class="op">$</span>labels</a>
<a class="sourceLine" id="cb10-3" data-line-number="3">true.baron.labels &lt;-<span class="st"> </span>baron<span class="op">@</span>colData<span class="op">@</span>listData[[<span class="st">&quot;label&quot;</span>]]</a>
<a class="sourceLine" id="cb10-4" data-line-number="4">unassigned_baron &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file=</span><span class="st">&quot;data/baron_unassigned_scClassify.rds&quot;</span>) </a>
<a class="sourceLine" id="cb10-5" data-line-number="5">pred.baron.labels &lt;-<span class="st"> </span>pred.baron.labels[<span class="op">!</span>unassigned_baron]</a>
<a class="sourceLine" id="cb10-6" data-line-number="6">pred.baron.labels &lt;-<span class="st"> </span>pred.baron.labels[<span class="op">!</span><span class="kw">is.na</span>(pred.baron.labels)]</a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="kw">saveRDS</span>(pred.baron.labels, <span class="dt">file=</span><span class="st">&quot;data/singleR.pred.baron.labels.rds&quot;</span>)</a>
<a class="sourceLine" id="cb10-8" data-line-number="8">true.baron.labels &lt;-<span class="st"> </span>true.baron.labels[<span class="op">!</span>unassigned_baron]</a>
<a class="sourceLine" id="cb10-9" data-line-number="9">true.baron.labels &lt;-<span class="st"> </span>true.baron.labels[<span class="op">!</span><span class="kw">is.na</span>(true.baron.labels)]</a>
<a class="sourceLine" id="cb10-10" data-line-number="10"><span class="kw">saveRDS</span>(true.baron.labels, <span class="dt">file=</span><span class="st">&quot;data/singleR.true.baron.labels.rds&quot;</span>)</a>
<a class="sourceLine" id="cb10-11" data-line-number="11"></a>
<a class="sourceLine" id="cb10-12" data-line-number="12"><span class="co">#Muraro: extract true and predicted labels</span></a>
<a class="sourceLine" id="cb10-13" data-line-number="13">pred.muraro.labels &lt;-<span class="st"> </span>pred.muraro<span class="op">$</span>labels</a>
<a class="sourceLine" id="cb10-14" data-line-number="14">true.muraro.labels &lt;-<span class="st"> </span>muraro<span class="op">@</span>colData<span class="op">@</span>listData[[<span class="st">&quot;label&quot;</span>]]</a>
<a class="sourceLine" id="cb10-15" data-line-number="15">unassigned_muraro &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file=</span><span class="st">&quot;data/muraro_unassigned_scClassify.rds&quot;</span>)</a>
<a class="sourceLine" id="cb10-16" data-line-number="16">pred.muraro.labels &lt;-<span class="st"> </span>pred.muraro.labels[<span class="op">!</span>unassigned_muraro]</a>
<a class="sourceLine" id="cb10-17" data-line-number="17">pred.muraro.labels &lt;-<span class="st"> </span>pred.muraro.labels[<span class="op">!</span><span class="kw">is.na</span>(pred.muraro.labels)]</a>
<a class="sourceLine" id="cb10-18" data-line-number="18"><span class="kw">saveRDS</span>(pred.muraro.labels, <span class="dt">file=</span><span class="st">&quot;data/singleR.pred.muraro.labels.rds&quot;</span>)</a>
<a class="sourceLine" id="cb10-19" data-line-number="19">true.muraro.labels &lt;-<span class="st"> </span>true.muraro.labels[<span class="op">!</span>unassigned_muraro]</a>
<a class="sourceLine" id="cb10-20" data-line-number="20">true.muraro.labels &lt;-<span class="st"> </span>true.muraro.labels[<span class="op">!</span><span class="kw">is.na</span>(true.muraro.labels)]</a>
<a class="sourceLine" id="cb10-21" data-line-number="21"><span class="kw">saveRDS</span>(true.muraro.labels, <span class="dt">file=</span><span class="st">&quot;data/singleR.true.muraro.labels.rds&quot;</span>)</a></code></pre></div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs">
<div id="ref-aran2019reference">
<p>Aran, Dvir, Agnieszka P Looney, Leqian Liu, Esther Wu, Valerie Fong, Austin Hsu, Suzanna Chak, et al. 2019. “Reference-Based Analysis of Lung Single-Cell Sequencing Reveals a Transitional Profibrotic Macrophage.” <em>Nature Immunology</em> 20 (2). Nature Publishing Group: 163–72.</p>
</div>
<div id="ref-singleRbook">
<p>Lun, Aaron. 2020. “Assigning Cell Types with Singler.” http://bioconductor.org/books/devel/SingleRBook/.</p>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] de_CH.UTF-8/de_CH.UTF-8/de_CH.UTF-8/C/de_CH.UTF-8/de_CH.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ggpubr_0.4.0                scuttle_1.0.3              
 [3] knitr_1.30                  scater_1.18.3              
 [5] ggplot2_3.3.2               ensembldb_2.14.0           
 [7] AnnotationFilter_1.14.0     GenomicFeatures_1.42.1     
 [9] AnnotationDbi_1.52.0        scRNAseq_2.4.0             
[11] SingleCellExperiment_1.12.0 SingleR_1.4.0              
[13] SummarizedExperiment_1.20.0 Biobase_2.50.0             
[15] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
[17] IRanges_2.24.0              S4Vectors_0.28.0           
[19] BiocGenerics_0.36.0         MatrixGenerics_1.2.0       
[21] matrixStats_0.57.0          workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] readxl_1.3.1                  backports_1.2.1              
  [3] AnnotationHub_2.22.0          BiocFileCache_1.14.0         
  [5] lazyeval_0.2.2                BiocParallel_1.24.1          
  [7] digest_0.6.27                 htmltools_0.5.0              
  [9] viridis_0.5.1                 magrittr_2.0.1               
 [11] memoise_1.1.0                 openxlsx_4.2.3               
 [13] Biostrings_2.58.0             askpass_1.1                  
 [15] prettyunits_1.1.1             colorspace_2.0-0             
 [17] blob_1.2.1                    rappdirs_0.3.1               
 [19] haven_2.3.1                   xfun_0.19                    
 [21] dplyr_1.0.2                   crayon_1.3.4                 
 [23] RCurl_1.98-1.2                glue_1.4.2                   
 [25] gtable_0.3.0                  zlibbioc_1.36.0              
 [27] XVector_0.30.0                DelayedArray_0.16.0          
 [29] car_3.0-10                    BiocSingular_1.6.0           
 [31] abind_1.4-5                   scales_1.1.1                 
 [33] DBI_1.1.0                     rstatix_0.6.0                
 [35] Rcpp_1.0.5                    viridisLite_0.3.0            
 [37] xtable_1.8-4                  progress_1.2.2               
 [39] foreign_0.8-80                bit_4.0.4                    
 [41] rsvd_1.0.3                    httr_1.4.2                   
 [43] ellipsis_0.3.1                farver_2.0.3                 
 [45] pkgconfig_2.0.3               XML_3.99-0.5                 
 [47] dbplyr_2.0.0                  labeling_0.4.2               
 [49] tidyselect_1.1.0              rlang_0.4.9                  
 [51] later_1.1.0.1                 munsell_0.5.0                
 [53] BiocVersion_3.12.0            cellranger_1.1.0             
 [55] tools_4.0.3                   generics_0.1.0               
 [57] RSQLite_2.2.1                 ExperimentHub_1.16.0         
 [59] broom_0.7.2                   evaluate_0.14                
 [61] stringr_1.4.0                 fastmap_1.0.1                
 [63] yaml_2.2.1                    bit64_4.0.5                  
 [65] fs_1.5.0                      zip_2.1.1                    
 [67] purrr_0.3.4                   sparseMatrixStats_1.2.0      
 [69] whisker_0.4                   mime_0.9                     
 [71] xml2_1.3.2                    biomaRt_2.46.0               
 [73] compiler_4.0.3                rstudioapi_0.13              
 [75] beeswarm_0.2.3                curl_4.3                     
 [77] interactiveDisplayBase_1.28.0 ggsignif_0.6.0               
 [79] tibble_3.0.4                  stringi_1.5.3                
 [81] highr_0.8                     forcats_0.5.0                
 [83] lattice_0.20-41               ProtGenerics_1.22.0          
 [85] Matrix_1.2-18                 vctrs_0.3.6                  
 [87] pillar_1.4.7                  lifecycle_0.2.0              
 [89] BiocManager_1.30.10           BiocNeighbors_1.8.2          
 [91] cowplot_1.1.0                 data.table_1.13.4            
 [93] bitops_1.0-6                  irlba_2.3.3                  
 [95] httpuv_1.5.4                  rtracklayer_1.50.0           
 [97] R6_2.5.0                      promises_1.1.1               
 [99] gridExtra_2.3                 rio_0.5.16                   
[101] vipor_0.4.5                   assertthat_0.2.1             
[103] openssl_1.4.3                 rprojroot_2.0.2              
[105] withr_2.3.0                   GenomicAlignments_1.26.0     
[107] Rsamtools_2.6.0               GenomeInfoDbData_1.2.4       
[109] hms_0.5.3                     grid_4.0.3                   
[111] beachmat_2.6.2                tidyr_1.1.2                  
[113] rmarkdown_2.5                 DelayedMatrixStats_1.12.1    
[115] carData_3.0-4                 git2r_0.27.1                 
[117] shiny_1.5.0                   ggbeeswarm_0.6.0             </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
